--- 
title: "Marine Genomics"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
description: ""
---

# Introduction to Marine Genomics

Placeholder



<!--chapter:end:index.Rmd-->


# Week 1- Welcome!

Placeholder


## Introduction to shell computing via the data carpentry tutorial
## How to access the shell
## Navigating your file system
## Shortcut: Tab Completion
## Summary & Key Points
## Navigating Files and Directories
## Week 1 Objectives
## Moving around the file system
## Examining the contents of other directories
## Full vs Relative Paths
## Navigational shortcuts
## Key Points
## Creature of the Week!

<!--chapter:end:01-Week1.Rmd-->


# Week 2-Working With Files

Placeholder


## Our data set: FASTQ files
## Wildcards
## Command History
## Examining Files
## Details on the FASTQ format
## Creating, moving, copying, and removing
### Copying Files
### Creating Directories
### Moving / Renaming 
### File Permissions
### Removing
## Redirections
## Searching files
## Redirecting output
## Writing for loops
## Using Basename in for loops
## Writing Scripts and Working with Data
## Writing files
## Writing scripts
## Moving and Downloading Data
### Getting data from the cloud
### Uploading and Downloading Data to your Virtual Machine with scp - UNIX
## Awk
## Creature of the Week!

<!--chapter:end:02-Week2.Rmd-->


# Week 3- What is a Genetic Variant?

Placeholder


## To get started lets dowload the data and install a few programs
## Raw read quality control
## Trimming to remove adapters
## Building an index of our genome
## Map reads to the genome
## sam to bam file conversion
## Genotype likelihoods
## Creature of the Week!

<!--chapter:end:03-MarGeno_week3.Rmd-->


# Week 4- The R environment

Placeholder


## Lesson 1: Orientation to R
## Manipulating a vector object 
## Operations act on each element of a vector:
## Operations can also work with two vectors:
## A few tips below for working with objects:
## Practice R Operations
## 1.2 Characterizing a dataframe
## How to access parts of the data:
## Data Manipulation
## Practice exploring a dataframe
## 1.3 Subsetting datasets & logicals
## Practice Subsetting datasets/logicals
## Creature of the Week!

<!--chapter:end:Marine_genomics_week4.rmd-->


# Week 5- R Continued

Placeholder


## 2.1 Plotting 
## Scatterplots
## Customizing your plot
## Practice Problems 2.1
## 2.2 plotting with ggplot2 
## Practice Problems 2.2
## For loops and the apply family of functions
## apply family
### sapply
### apply 
### lapply -- "list" apply
### tapply - "per Type" apply 
## Exercise 2.3 apply and tapply
## Creature of the Week!

<!--chapter:end:Week5_R_Introduction_continued.Rmd-->


# Week 6: PCA 

Placeholder


## Download the data
## Installing programs
## Run pcangsd on our data
## Setting up a new project in R
## Reading data into R
## Exercises
## Creature of the Week!

<!--chapter:end:Week6.Rmd-->


# Week 7: F<sub>st</sub> and outlier analysis

Placeholder


## Download the data
## Getting R set up
## Finding outliers using pcadapt
## pcadapt Exercises
## Using F<sub>st</sub> to find outliers
## OutFLANK Practice
## Creature of the Week!

<!--chapter:end:Week7.Rmd-->

---
title: "Week8_GWAS"
author: "Serena Caplins"
date: "5/12/2021"
output: html_document
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/Desktop/MarineGenomics-main/MarineGenomics/data/Week8/MarineGenomicsData/Week8")
```

# Week 8: Genome wide association study (GWAS)

This week we're going to show you how to perform a genome wide association study or GWAS.  

The lecture for this week can be found here.


## load in the data 

The data for this week consist of several vcf files in beagle format, a genome index file (.fai), and a few small text files.

```html

wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/week8.tar.gz


tar -xzvf week8.tar.gz

#and one more vcf file that was too big to upload with the others

cd MarineGenomicsData/week8

wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/salmon_chr6_19ind.BEAGLE.PL.gz

```

## install packages in R

We'll be using the package `qqman` to make a manhattan plot.

```{r, echo=T}

#install.package("qqman")

```

## install angsd and pcangsd (again?)

If you missed week 6 and didn't get pcangsd or angsd(week 3) installed. Here are the scripts to install them.They take a few minutes to install. There is no need to install these again if you installed them last time. 

```html
## install angsd

cd
  git clone --recursive https://github.com/samtools/htslib.git
  git clone https://github.com/ANGSD/angsd.git 
  cd htslib;make;cd ../angsd ;make HTSSRC=../htslib
   
cd

##for pcangsd

#install pip for python
curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
python3 get-pip.py

#we need to add our home directory to the path for pip
#look at path
echo $PATH

#add the location of pip to our path
export PATH="$HOME/.local/bin:$PATH"


# then install pcangsd
git clone https://github.com/Rosemeis/pcangsd.git
cd pcangsd/
pip install --user -r requirements.txt #this installs additional requirements for pcangsd
python3 setup.py build_ext --inplace
```

## The data

The data we're working on for the GWAS comes from the paper by Kijas et al. 2018 where they studied the genetic basis of sex determination in the atlantic Salmon. The paper can be found [here](https://www.nature.com/articles/s41598-018-23984-1#Sec2). In short they examined the genetic basis of sex in 19 salmon for which they have whole genome sequence data. We'll only be looking at two chromosomes (2 and 6) of this data.

## take VCF file and generate lrt file

To do the test of genome wide association we need to take our vcf file and test whether there is an association with our phenotype (in this case whether a fish has a male or female phenotype). The phenotypes are coded as 0 = Female and 1 = Male in the `phenobin` file. 

```html
$HOME/angsd/angsd -doMaf 4 -beagle salmon_chr2_19ind.BEAGLE.PL.gz -yBin phenobin -doAsso 2 -fai Salmon.fai

```

This will generate several output files labeled `angsdput`. We'll use the file with the lrt0 extension to plot our manattan plot.

## take lrt file and make a manhattan plot

```{r, eval=F}
#in R
#set working directory

setwd("~/Marinegenomics/Week_8 GWAS/MarineGenomicsData/Week8/")

#read in the data
lrt<-read.table(gzfile("angsdput.lrt0.gz"), header=T, sep="\t")

#look at it
str(lrt)

#we have a few LRT values that are -999, we should remove them. How many do we have?
length(which(lrt$LRT == -999))
#210597 #most are filtered out

length(lrt$LRT)
#224460

length(lrt$LRT)-length(which(lrt$LRT == -999))
#[1] 13863


#remove the values that are not -999 and that are negative

lrt_filt<-lrt[-c(which(lrt$LRT == -999),which(lrt$LRT <= 0)),]


hist(lrt_filt$LRT, breaks=50)
```
```{r, include=F}
#read in the data
lrt<-read.table(gzfile("angsdput.lrt0.gz"), header=T, sep="\t")

#look at it
str(lrt)

#we have a few LRT values that are -999, we should remove them. How many do we have?
length(which(lrt$LRT == -999))
#210597 #most are filtered out

length(lrt$LRT)
#224460

length(lrt$LRT)-length(which(lrt$LRT == -999))
#[1] 13863


#remove the values that are not -999 and that are negative

lrt_filt<-lrt[-c(which(lrt$LRT == -999),which(lrt$LRT <= 0)),]


hist(lrt_filt$LRT, breaks=50)
```


Everything looks good to proceed to making our manhattan plot.

```{r, echo=T}

require(qqman)

#the function manhattan requires each SNP to have it's own "name". Let's make a vector for rownumbers that start with the letter r.

lrt_filt$SNP<-paste("r",1:length(lrt_filt$Chromosome), sep="")

#we also need to make sure we don't have any tricky values like those below
lrt_filt<-lrt_filt[-c(which(lrt_filt$pvalue == "NaN" ),
                      which(lrt_filt$pvalue == "Inf"),
                      which(lrt_filt$LRT == "Inf")),]


manhattan(lrt_filt, chr="Chromosome", bp="Position", logp=F, p="LRT")


```

We can highlight the values that are exceed a threshold. There are several ways to determine a threshold, but one is to make a vector of random phenotypes and re-run our association test. We can then set the highest LRT value from the random phenotype test as our upper limit for our plot with the actual phenotypes.


```{r, echo=T}
#make a vector with 19 1's and 0's

x<-sample(c(1,0), 19, replace=T)

#write this to our week 8 directory

write.table(x, "rando_pheno", row.names = F)

```

And now use that phenotype file to run our association test again, making sure to specify a different output file.

```html

$HOME/angsd/angsd -doMaf 4 -beagle salmon_chr2_19ind.BEAGLE.PL.gz -yBin rando_pheno -doAsso 2 -fai Salmon.fai -out randotest

```


And rerun the code in R to see what our maximum LRT values are in this random phenotype test.

```{r, echo=T}

lrt_rando<-read.table(gzfile("MarineGenomicsData/Week8/randotest.lrt0.gz"), header=T, sep="\t")

#we need to remove those -999 values again

rando_filt<-lrt_rando[-c(which(lrt_rando$LRT == -999),which(lrt_rando$LRT <= 0)),]

summary(rando_filt$LRT, na.rm=T)

#we have some Inf values we also need to remove, let's add those to our filtering line above

rando_filt<-lrt_rando[-c(which(lrt_rando$LRT == -999),which(lrt_rando$LRT <= 0), which(lrt_rando$LRT == Inf)),]

max(rando_filt$LRT, na.rm=T)


```

So we can highlight all of the SNPs that have an LRT greater than 12 in our association test.


```{r, echo=T}

#make a list of the candidates
candidates<-lrt_filt[which(lrt_filt$LRT > 12.33),]$SNP


#refer to that list of candidates with the highlight parameter
manhattan(lrt_filt, chr="Chromosome", bp="Position", logp=F, p="LRT", highlight = candidates)


```
Comparing our results to the Kijas et al. 2018 paper, we have a similar pattern of many SNPs across the chromosome showing a relationship with phenotype (sex). Interestingly this paper found that there are three chromosomes that are associated with sex in Atlantic Salmon, but not all chromosomes give a strong signal in all individuals. For example, only three male individuals were found to have a clear association with chromosome 2, and the other males in the study were found to have an association with chromosomes 3 and 6. 

These results highlight the fluid nature of sex determination in animals, even those with a genetic basis to sex determination.

For the exercise you'll take a closer look at chromosome 6, where you'll try to find the individuals that are male from a PCA plot.

## Exercises

> 1.For this exercise, we want to see which individuals actually show genomic levels of variation at chromosome 3. Using the code that we ran for week 6. Make a PCA plot of the salmon_chr3_19ind.BEAGLE.PL.gz file. Use the function `indentify()` to find the points that are clustering apart from the other points. Verify that these are males by referring to the table in the Kijas et al. paper [here](https://www.nature.com/articles/s41598-018-23984-1/tables/1). Note their individuals are in the same order as our samples though the names don't match.

<details><summary><span style="color: orange;">Solution</span></summary>
<p>

```html
#run pcangsd on our chr3 data
python3 ../../pcangsd/pcangsd.py -beagle salmon_chr3_19ind.BEAGLE.PL.gz -o pca6_out -threads 28

```
In R make a PCA plot

```{r, echo=T}
#read in the data
cov<-as.matrix(read.table("MarineGenomicsData/Week8/pca6_out.cov"))

#compute the eigen values

e<-eigen(cov)

#how much variation are we explaining here?
e$values/sum(e$values)

#looks like 13.7% on the first axis

#make a plot of the first two axes
plot(e$vectors[,1:2])

#use identify to find the points that are clustered apart from the others

#identify(e$vectors[,1:2]) This doesn't work in R studio so we'll show the plot below

#If you select the points on the left most of the screen, you will find they are rows
# 1, 2, and 12

#If you select the other two points that are leftmost from the remaining points you also get
# 16 and 19

```

```{r echo=F}

knitr::include_graphics("./fig/Rplot_PCACh6.png")
```

Looking at the table in the Kilas paper, the Animals in rows 1, 2, and 12 are Male, but it's rows 17 and 18 that are next not 16 and 19, so we may have somethings out of order in our list.


</p>
</details>
&nbsp;


> 2. Make a manhattan plot for chromosome 6. You will need to make a new phenotype file that has 1's for all the males identified in problem 1. Alterntatively, you can make a phenotype file from table 1 in Kilas et al. 2018, setting the animals that were found to be male on chr 6 to 1 and all other animals to 0.

<!--chapter:end:Week8Markdown.rmd-->

