<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Week 8: Genome wide association study (GWAS) | Week8_GWAS</title>
  <meta name="description" content="9 Week 8: Genome wide association study (GWAS) | Week8_GWAS" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Week 8: Genome wide association study (GWAS) | Week8_GWAS" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Week 8: Genome wide association study (GWAS) | Week8_GWAS" />
  
  
  

<meta name="author" content="Serena Caplins" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="week-7-fst-and-outlier-analysis.html"/>
<link rel="next" href="week-9-population-structure-using-ngsadmix.html"/>
<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Marine Genomics 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to Marine Genomics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#cloud-computing-versus-individual-laptops"><i class="fa fa-check"></i><b>1.1</b> Cloud computing versus individual laptops</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#meet-our-instructors"><i class="fa fa-check"></i><b>1.2</b> Meet our instructors!</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.3</b> Schedule</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="week-1-welcome.html"><a href="week-1-welcome.html"><i class="fa fa-check"></i><b>2</b> Week 1- Welcome!</a>
<ul>
<li class="chapter" data-level="2.1" data-path="week-1-welcome.html"><a href="week-1-welcome.html#introduction-to-shell-computing-via-the-data-carpentry-tutorial"><i class="fa fa-check"></i><b>2.1</b> Introduction to shell computing via the data carpentry tutorial</a></li>
<li class="chapter" data-level="2.2" data-path="week-1-welcome.html"><a href="week-1-welcome.html#how-to-access-the-shell"><i class="fa fa-check"></i><b>2.2</b> How to access the shell</a></li>
<li class="chapter" data-level="2.3" data-path="week-1-welcome.html"><a href="week-1-welcome.html#navigating-your-file-system"><i class="fa fa-check"></i><b>2.3</b> Navigating your file system</a></li>
<li class="chapter" data-level="2.4" data-path="week-1-welcome.html"><a href="week-1-welcome.html#shortcut-tab-completion"><i class="fa fa-check"></i><b>2.4</b> Shortcut: Tab Completion</a></li>
<li class="chapter" data-level="2.5" data-path="week-1-welcome.html"><a href="week-1-welcome.html#summary-key-points"><i class="fa fa-check"></i><b>2.5</b> Summary &amp; Key Points</a></li>
<li class="chapter" data-level="2.6" data-path="week-1-welcome.html"><a href="week-1-welcome.html#navigating-files-and-directories"><i class="fa fa-check"></i><b>2.6</b> Navigating Files and Directories</a></li>
<li class="chapter" data-level="2.7" data-path="week-1-welcome.html"><a href="week-1-welcome.html#week-1-objectives"><i class="fa fa-check"></i><b>2.7</b> Week 1 Objectives</a></li>
<li class="chapter" data-level="2.8" data-path="week-1-welcome.html"><a href="week-1-welcome.html#moving-around-the-file-system"><i class="fa fa-check"></i><b>2.8</b> Moving around the file system</a></li>
<li class="chapter" data-level="2.9" data-path="week-1-welcome.html"><a href="week-1-welcome.html#examining-the-contents-of-other-directories"><i class="fa fa-check"></i><b>2.9</b> Examining the contents of other directories</a></li>
<li class="chapter" data-level="2.10" data-path="week-1-welcome.html"><a href="week-1-welcome.html#full-vs-relative-paths"><i class="fa fa-check"></i><b>2.10</b> Full vs Relative Paths</a></li>
<li class="chapter" data-level="2.11" data-path="week-1-welcome.html"><a href="week-1-welcome.html#navigational-shortcuts"><i class="fa fa-check"></i><b>2.11</b> Navigational shortcuts</a></li>
<li class="chapter" data-level="2.12" data-path="week-1-welcome.html"><a href="week-1-welcome.html#key-points"><i class="fa fa-check"></i><b>2.12</b> Key Points</a></li>
<li class="chapter" data-level="2.13" data-path="week-1-welcome.html"><a href="week-1-welcome.html#creature-of-the-week"><i class="fa fa-check"></i><b>2.13</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html"><i class="fa fa-check"></i><b>3</b> Week 2-Working With Files</a>
<ul>
<li class="chapter" data-level="3.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#our-data-set-fastq-files"><i class="fa fa-check"></i><b>3.1</b> Our data set: FASTQ files</a></li>
<li class="chapter" data-level="3.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#wildcards"><i class="fa fa-check"></i><b>3.2</b> Wildcards</a></li>
<li class="chapter" data-level="3.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#command-history"><i class="fa fa-check"></i><b>3.3</b> Command History</a></li>
<li class="chapter" data-level="3.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#examining-files"><i class="fa fa-check"></i><b>3.4</b> Examining Files</a></li>
<li class="chapter" data-level="3.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#details-on-the-fastq-format"><i class="fa fa-check"></i><b>3.5</b> Details on the FASTQ format</a></li>
<li class="chapter" data-level="3.6" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-moving-copying-and-removing"><i class="fa fa-check"></i><b>3.6</b> Creating, moving, copying, and removing</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#copying-files"><i class="fa fa-check"></i><b>3.6.1</b> Copying Files</a></li>
<li class="chapter" data-level="3.6.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-directories"><i class="fa fa-check"></i><b>3.6.2</b> Creating Directories</a></li>
<li class="chapter" data-level="3.6.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#moving-renaming"><i class="fa fa-check"></i><b>3.6.3</b> Moving / Renaming</a></li>
<li class="chapter" data-level="3.6.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#file-permissions"><i class="fa fa-check"></i><b>3.6.4</b> File Permissions</a></li>
<li class="chapter" data-level="3.6.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#removing"><i class="fa fa-check"></i><b>3.6.5</b> Removing</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#redirections"><i class="fa fa-check"></i><b>3.7</b> Redirections</a></li>
<li class="chapter" data-level="3.8" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#searching-files"><i class="fa fa-check"></i><b>3.8</b> Searching files</a></li>
<li class="chapter" data-level="3.9" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#redirecting-output"><i class="fa fa-check"></i><b>3.9</b> Redirecting output</a></li>
<li class="chapter" data-level="3.10" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-for-loops"><i class="fa fa-check"></i><b>3.10</b> Writing for loops</a></li>
<li class="chapter" data-level="3.11" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#using-basename-in-for-loops"><i class="fa fa-check"></i><b>3.11</b> Using Basename in for loops</a></li>
<li class="chapter" data-level="3.12" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts-and-working-with-data"><i class="fa fa-check"></i><b>3.12</b> Writing Scripts and Working with Data</a></li>
<li class="chapter" data-level="3.13" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-files"><i class="fa fa-check"></i><b>3.13</b> Writing files</a></li>
<li class="chapter" data-level="3.14" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts"><i class="fa fa-check"></i><b>3.14</b> Writing scripts</a></li>
<li class="chapter" data-level="3.15" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#moving-and-downloading-data"><i class="fa fa-check"></i><b>3.15</b> Moving and Downloading Data</a>
<ul>
<li class="chapter" data-level="3.15.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#getting-data-from-the-cloud"><i class="fa fa-check"></i><b>3.15.1</b> Getting data from the cloud</a></li>
<li class="chapter" data-level="3.15.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix"><i class="fa fa-check"></i><b>3.15.2</b> Uploading and Downloading Data to your Virtual Machine with scp - UNIX</a></li>
</ul></li>
<li class="chapter" data-level="3.16" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#awk"><i class="fa fa-check"></i><b>3.16</b> Awk</a></li>
<li class="chapter" data-level="3.17" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creature-of-the-week-1"><i class="fa fa-check"></i><b>3.17</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html"><i class="fa fa-check"></i><b>4</b> Week 3- What is a Genetic Variant?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#to-get-started-lets-dowload-the-data-and-install-a-few-programs"><i class="fa fa-check"></i><b>4.1</b> To get started lets dowload the data and install a few programs</a></li>
<li class="chapter" data-level="4.2" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#raw-read-quality-control"><i class="fa fa-check"></i><b>4.2</b> Raw read quality control</a></li>
<li class="chapter" data-level="4.3" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#trimming-to-remove-adapters"><i class="fa fa-check"></i><b>4.3</b> Trimming to remove adapters</a></li>
<li class="chapter" data-level="4.4" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#building-an-index-of-our-genome"><i class="fa fa-check"></i><b>4.4</b> Building an index of our genome</a></li>
<li class="chapter" data-level="4.5" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#map-reads-to-the-genome"><i class="fa fa-check"></i><b>4.5</b> Map reads to the genome</a></li>
<li class="chapter" data-level="4.6" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#sam-to-bam-file-conversion"><i class="fa fa-check"></i><b>4.6</b> sam to bam file conversion</a></li>
<li class="chapter" data-level="4.7" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#genotype-likelihoods"><i class="fa fa-check"></i><b>4.7</b> Genotype likelihoods</a></li>
<li class="chapter" data-level="4.8" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#creature-of-the-week-2"><i class="fa fa-check"></i><b>4.8</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html"><i class="fa fa-check"></i><b>5</b> Week 4- The R environment</a>
<ul>
<li class="chapter" data-level="5.1" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#lesson-1-orientation-to-r"><i class="fa fa-check"></i><b>5.1</b> Lesson 1: Orientation to R</a></li>
<li class="chapter" data-level="5.2" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#manipulating-a-vector-object"><i class="fa fa-check"></i><b>5.2</b> Manipulating a vector object</a></li>
<li class="chapter" data-level="5.3" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#operations-act-on-each-element-of-a-vector"><i class="fa fa-check"></i><b>5.3</b> Operations act on each element of a vector:</a></li>
<li class="chapter" data-level="5.4" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#operations-can-also-work-with-two-vectors"><i class="fa fa-check"></i><b>5.4</b> Operations can also work with two vectors:</a></li>
<li class="chapter" data-level="5.5" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#a-few-tips-below-for-working-with-objects"><i class="fa fa-check"></i><b>5.5</b> A few tips below for working with objects:</a></li>
<li class="chapter" data-level="5.6" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#practice-r-operations"><i class="fa fa-check"></i><b>5.6</b> Practice R Operations</a></li>
<li class="chapter" data-level="5.7" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#characterizing-a-dataframe"><i class="fa fa-check"></i><b>5.7</b> 1.2 Characterizing a dataframe</a></li>
<li class="chapter" data-level="5.8" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#how-to-access-parts-of-the-data"><i class="fa fa-check"></i><b>5.8</b> How to access parts of the data:</a></li>
<li class="chapter" data-level="5.9" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#data-manipulation"><i class="fa fa-check"></i><b>5.9</b> Data Manipulation</a></li>
<li class="chapter" data-level="5.10" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#practice-exploring-a-dataframe"><i class="fa fa-check"></i><b>5.10</b> Practice exploring a dataframe</a></li>
<li class="chapter" data-level="5.11" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#subsetting-datasets-logicals"><i class="fa fa-check"></i><b>5.11</b> 1.3 Subsetting datasets &amp; logicals</a></li>
<li class="chapter" data-level="5.12" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#practice-subsetting-datasetslogicals"><i class="fa fa-check"></i><b>5.12</b> Practice Subsetting datasets/logicals</a></li>
<li class="chapter" data-level="5.13" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#creature-of-the-week-3"><i class="fa fa-check"></i><b>5.13</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html"><i class="fa fa-check"></i><b>6</b> Week 5- R Continued</a>
<ul>
<li class="chapter" data-level="6.1" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#plotting"><i class="fa fa-check"></i><b>6.1</b> 2.1 Plotting</a></li>
<li class="chapter" data-level="6.2" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#scatterplots"><i class="fa fa-check"></i><b>6.2</b> Scatterplots</a></li>
<li class="chapter" data-level="6.3" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#customizing-your-plot"><i class="fa fa-check"></i><b>6.3</b> Customizing your plot</a></li>
<li class="chapter" data-level="6.4" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#practice-problems-2.1"><i class="fa fa-check"></i><b>6.4</b> Practice Problems 2.1</a></li>
<li class="chapter" data-level="6.5" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#plotting-with-ggplot2"><i class="fa fa-check"></i><b>6.5</b> 2.2 plotting with ggplot2</a></li>
<li class="chapter" data-level="6.6" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#practice-problems-2.2"><i class="fa fa-check"></i><b>6.6</b> Practice Problems 2.2</a></li>
<li class="chapter" data-level="6.7" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#for-loops-and-the-apply-family-of-functions"><i class="fa fa-check"></i><b>6.7</b> For loops and the apply family of functions</a></li>
<li class="chapter" data-level="6.8" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#apply-family"><i class="fa fa-check"></i><b>6.8</b> apply family</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#sapply"><i class="fa fa-check"></i><b>6.8.1</b> sapply</a></li>
<li class="chapter" data-level="6.8.2" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#apply"><i class="fa fa-check"></i><b>6.8.2</b> apply</a></li>
<li class="chapter" data-level="6.8.3" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#lapply-list-apply"><i class="fa fa-check"></i><b>6.8.3</b> lapply – “list” apply</a></li>
<li class="chapter" data-level="6.8.4" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#tapply---per-type-apply"><i class="fa fa-check"></i><b>6.8.4</b> tapply - “per Type” apply</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#exercise-2.3-apply-and-tapply"><i class="fa fa-check"></i><b>6.9</b> Exercise 2.3 apply and tapply</a></li>
<li class="chapter" data-level="6.10" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#creature-of-the-week-4"><i class="fa fa-check"></i><b>6.10</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="week-6-principle-component-analyses.html"><a href="week-6-principle-component-analyses.html"><i class="fa fa-check"></i><b>7</b> Week 6- Principle Component Analyses</a>
<ul>
<li class="chapter" data-level="7.1" data-path="week-6-principle-component-analyses.html"><a href="week-6-principle-component-analyses.html#download-the-data"><i class="fa fa-check"></i><b>7.1</b> Download the data</a></li>
<li class="chapter" data-level="7.2" data-path="week-6-principle-component-analyses.html"><a href="week-6-principle-component-analyses.html#installing-programs"><i class="fa fa-check"></i><b>7.2</b> Installing programs</a></li>
<li class="chapter" data-level="7.3" data-path="week-6-principle-component-analyses.html"><a href="week-6-principle-component-analyses.html#run-pcangsd-on-our-data"><i class="fa fa-check"></i><b>7.3</b> Run pcangsd on our data</a></li>
<li class="chapter" data-level="7.4" data-path="week-6-principle-component-analyses.html"><a href="week-6-principle-component-analyses.html#setting-up-a-new-project-in-r"><i class="fa fa-check"></i><b>7.4</b> Setting up a new project in R</a></li>
<li class="chapter" data-level="7.5" data-path="week-6-principle-component-analyses.html"><a href="week-6-principle-component-analyses.html#reading-data-into-r"><i class="fa fa-check"></i><b>7.5</b> Reading data into R</a></li>
<li class="chapter" data-level="7.6" data-path="week-6-principle-component-analyses.html"><a href="week-6-principle-component-analyses.html#exercises"><i class="fa fa-check"></i><b>7.6</b> Exercises</a></li>
<li class="chapter" data-level="7.7" data-path="week-6-principle-component-analyses.html"><a href="week-6-principle-component-analyses.html#creature-of-the-week-5"><i class="fa fa-check"></i><b>7.7</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html"><i class="fa fa-check"></i><b>8</b> Week 7- F<sub>st</sub> and outlier analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#download-the-data-1"><i class="fa fa-check"></i><b>8.1</b> Download the data</a></li>
<li class="chapter" data-level="8.2" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#getting-r-set-up"><i class="fa fa-check"></i><b>8.2</b> Getting R set up</a></li>
<li class="chapter" data-level="8.3" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#finding-outliers-using-pcadapt"><i class="fa fa-check"></i><b>8.3</b> Finding outliers using pcadapt</a></li>
<li class="chapter" data-level="8.4" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#pcadapt-exercises"><i class="fa fa-check"></i><b>8.4</b> pcadapt Exercises</a></li>
<li class="chapter" data-level="8.5" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#using-fst-to-find-outliers"><i class="fa fa-check"></i><b>8.5</b> Using F<sub>st</sub> to find outliers</a></li>
<li class="chapter" data-level="8.6" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#outflank-practice"><i class="fa fa-check"></i><b>8.6</b> OutFLANK Practice</a></li>
<li class="chapter" data-level="8.7" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#creature-of-the-week-6"><i class="fa fa-check"></i><b>8.7</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html"><i class="fa fa-check"></i><b>9</b> Week 8: Genome wide association study (GWAS)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#load-in-the-data"><i class="fa fa-check"></i><b>9.1</b> load in the data</a></li>
<li class="chapter" data-level="9.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-packages-in-r"><i class="fa fa-check"></i><b>9.2</b> install packages in R</a></li>
<li class="chapter" data-level="9.3" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd-and-pcangsd-again"><i class="fa fa-check"></i><b>9.3</b> install angsd and pcangsd (again?)</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd"><i class="fa fa-check"></i><b>9.3.1</b> install angsd</a></li>
<li class="chapter" data-level="9.3.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#for-pcangsd"><i class="fa fa-check"></i><b>9.3.2</b> for pcangsd</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#the-data"><i class="fa fa-check"></i><b>9.4</b> The data</a></li>
<li class="chapter" data-level="9.5" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-beagle-file-and-generate-lrt-file"><i class="fa fa-check"></i><b>9.5</b> take Beagle file and generate lrt file</a></li>
<li class="chapter" data-level="9.6" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-lrt-file-and-make-a-manhattan-plot"><i class="fa fa-check"></i><b>9.6</b> take lrt file and make a manhattan plot</a></li>
<li class="chapter" data-level="9.7" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#exercises-1"><i class="fa fa-check"></i><b>9.7</b> Exercises</a></li>
<li class="chapter" data-level="9.8" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#creature-of-the-week-7"><i class="fa fa-check"></i><b>9.8</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="week-9-population-structure-using-ngsadmix.html"><a href="week-9-population-structure-using-ngsadmix.html"><i class="fa fa-check"></i><b>10</b> Week 9- Population Structure using NGSadmix</a>
<ul>
<li class="chapter" data-level="10.1" data-path="week-9-population-structure-using-ngsadmix.html"><a href="week-9-population-structure-using-ngsadmix.html#download-the-data-2"><i class="fa fa-check"></i><b>10.1</b> Download the data</a></li>
<li class="chapter" data-level="10.2" data-path="week-9-population-structure-using-ngsadmix.html"><a href="week-9-population-structure-using-ngsadmix.html#using-ngsadmix"><i class="fa fa-check"></i><b>10.2</b> Using NGSadmix</a></li>
<li class="chapter" data-level="10.3" data-path="week-9-population-structure-using-ngsadmix.html"><a href="week-9-population-structure-using-ngsadmix.html#ploting-stucture-in-r"><i class="fa fa-check"></i><b>10.3</b> ploting stucture in R</a></li>
<li class="chapter" data-level="10.4" data-path="week-9-population-structure-using-ngsadmix.html"><a href="week-9-population-structure-using-ngsadmix.html#how-do-we-know-which-k-to-pick"><i class="fa fa-check"></i><b>10.4</b> How do we know which K to pick?</a></li>
<li class="chapter" data-level="10.5" data-path="week-9-population-structure-using-ngsadmix.html"><a href="week-9-population-structure-using-ngsadmix.html#exercises-2"><i class="fa fa-check"></i><b>10.5</b> Exercises</a></li>
<li class="chapter" data-level="10.6" data-path="week-9-population-structure-using-ngsadmix.html"><a href="week-9-population-structure-using-ngsadmix.html#creature-of-the-week-8"><i class="fa fa-check"></i><b>10.6</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html"><i class="fa fa-check"></i><b>11</b> Week 10- Genomics is a BLAST</a>
<ul>
<li class="chapter" data-level="11.1" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html#genome-annotation-and-blast-searches"><i class="fa fa-check"></i><b>11.1</b> Genome Annotation and BLAST searches</a></li>
<li class="chapter" data-level="11.2" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html#download-the-data-3"><i class="fa fa-check"></i><b>11.2</b> Download the data</a></li>
<li class="chapter" data-level="11.3" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html#install-transdecoder"><i class="fa fa-check"></i><b>11.3</b> Install Transdecoder</a></li>
<li class="chapter" data-level="11.4" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html#run-transdecoder"><i class="fa fa-check"></i><b>11.4</b> Run Transdecoder</a></li>
<li class="chapter" data-level="11.5" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html#run-ncbi-blast"><i class="fa fa-check"></i><b>11.5</b> Run NCBI-BLAST</a></li>
<li class="chapter" data-level="11.6" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html#what-did-we-find"><i class="fa fa-check"></i><b>11.6</b> What did we find?</a></li>
<li class="chapter" data-level="11.7" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html#the-importance-of-genomic-architecture"><i class="fa fa-check"></i><b>11.7</b> The importance of Genomic Architecture</a></li>
<li class="chapter" data-level="11.8" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html#exercises-3"><i class="fa fa-check"></i><b>11.8</b> Exercises</a></li>
<li class="chapter" data-level="11.9" data-path="week-10-genomics-is-a-blast.html"><a href="week-10-genomics-is-a-blast.html#creature-of-the-week-9"><i class="fa fa-check"></i><b>11.9</b> Creature of the Week!</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Week8_GWAS</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="week-8-genome-wide-association-study-gwas" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Week 8: Genome wide association study (GWAS)</h1>
<p>This week we’re going to show you how to perform a genome wide association study or GWAS.</p>
<p>The lecture for this week can be found <a href="https://github.com/BayLab/MarineGenomics/blob/4f22a87ad800f98a6e138a166b8e94d016310914/ppt/Week8_GWAS.pdf">here</a></p>
<div id="load-in-the-data" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> load in the data</h2>
<p>The data for this week consist of several files in beagle format, a genome index file (.fai), and a few small text files.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb1-1"><a href="week-8-genome-wide-association-study-gwas.html#cb1-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2"><a href="week-8-genome-wide-association-study-gwas.html#cb1-2" aria-hidden="true" tabindex="-1"></a>wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/week8.tar.gz</span>
<span id="cb1-3"><a href="week-8-genome-wide-association-study-gwas.html#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="week-8-genome-wide-association-study-gwas.html#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="week-8-genome-wide-association-study-gwas.html#cb1-5" aria-hidden="true" tabindex="-1"></a>tar -xzvf week8.tar.gz</span>
<span id="cb1-6"><a href="week-8-genome-wide-association-study-gwas.html#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="week-8-genome-wide-association-study-gwas.html#cb1-7" aria-hidden="true" tabindex="-1"></a>#and one more file that was too big to upload with the others</span>
<span id="cb1-8"><a href="week-8-genome-wide-association-study-gwas.html#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="week-8-genome-wide-association-study-gwas.html#cb1-9" aria-hidden="true" tabindex="-1"></a>cd MarineGenomicsData/week8</span>
<span id="cb1-10"><a href="week-8-genome-wide-association-study-gwas.html#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="week-8-genome-wide-association-study-gwas.html#cb1-11" aria-hidden="true" tabindex="-1"></a>wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/salmon_chr6_19ind.BEAGLE.PL.gz</span></code></pre></div>
</div>
<div id="install-packages-in-r" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> install packages in R</h2>
<p>We’ll be using the package <code>qqman</code> to make a manhattan plot.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="week-8-genome-wide-association-study-gwas.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#install.package(&quot;qqman&quot;)</span></span></code></pre></div>
</div>
<div id="install-angsd-and-pcangsd-again" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> install angsd and pcangsd (again?)</h2>
<p>If you missed week 6 and didn’t get pcangsd or angsd(week 3) installed. Here are the scripts to install them.They take a few minutes to install. There is no need to install these again if you installed them last time.</p>
<div id="install-angsd" class="section level3" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> install angsd</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb3-1"><a href="week-8-genome-wide-association-study-gwas.html#cb3-1" aria-hidden="true" tabindex="-1"></a>cd</span>
<span id="cb3-2"><a href="week-8-genome-wide-association-study-gwas.html#cb3-2" aria-hidden="true" tabindex="-1"></a>  git clone --recursive https://github.com/samtools/htslib.git</span>
<span id="cb3-3"><a href="week-8-genome-wide-association-study-gwas.html#cb3-3" aria-hidden="true" tabindex="-1"></a>  git clone https://github.com/ANGSD/angsd.git </span>
<span id="cb3-4"><a href="week-8-genome-wide-association-study-gwas.html#cb3-4" aria-hidden="true" tabindex="-1"></a>  cd htslib;make;cd ../angsd ;make HTSSRC=../htslib</span>
<span id="cb3-5"><a href="week-8-genome-wide-association-study-gwas.html#cb3-5" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb3-6"><a href="week-8-genome-wide-association-study-gwas.html#cb3-6" aria-hidden="true" tabindex="-1"></a>cd</span></code></pre></div>
</div>
<div id="for-pcangsd" class="section level3" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> for pcangsd</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb4-1"><a href="week-8-genome-wide-association-study-gwas.html#cb4-1" aria-hidden="true" tabindex="-1"></a>install pip for python</span>
<span id="cb4-2"><a href="week-8-genome-wide-association-study-gwas.html#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="week-8-genome-wide-association-study-gwas.html#cb4-3" aria-hidden="true" tabindex="-1"></a>curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py</span>
<span id="cb4-4"><a href="week-8-genome-wide-association-study-gwas.html#cb4-4" aria-hidden="true" tabindex="-1"></a>python3 get-pip.py</span></code></pre></div>
<p>we need to add our home directory to the path for pip
look at path echo $PATH</p>
<p>add the location of pip to our path</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb5-1"><a href="week-8-genome-wide-association-study-gwas.html#cb5-1" aria-hidden="true" tabindex="-1"></a>export PATH=&quot;$HOME/.local/bin:$PATH&quot;</span></code></pre></div>
<p>then install pcangsd</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb6-1"><a href="week-8-genome-wide-association-study-gwas.html#cb6-1" aria-hidden="true" tabindex="-1"></a>git clone https://github.com/Rosemeis/pcangsd.git</span>
<span id="cb6-2"><a href="week-8-genome-wide-association-study-gwas.html#cb6-2" aria-hidden="true" tabindex="-1"></a>cd pcangsd/</span>
<span id="cb6-3"><a href="week-8-genome-wide-association-study-gwas.html#cb6-3" aria-hidden="true" tabindex="-1"></a>pip install --user -r requirements.txt #this installs additional requirements for pcangsd</span>
<span id="cb6-4"><a href="week-8-genome-wide-association-study-gwas.html#cb6-4" aria-hidden="true" tabindex="-1"></a>python3 setup.py build_ext --inplace</span></code></pre></div>
</div>
</div>
<div id="the-data" class="section level2" number="9.4">
<h2><span class="header-section-number">9.4</span> The data</h2>
<p>The data we’re working on for the GWAS comes from the paper by Kijas et al. 2018 where they studied the genetic basis of sex determination in the atlantic Salmon. The paper can be found <a href="https://www.nature.com/articles/s41598-018-23984-1#Sec2">here</a>. In short they examined the genetic basis of sex in 19 salmon for which they have whole genome sequence data. We’ll only be looking at two chromosomes (2 and 6) of this data.</p>
</div>
<div id="take-beagle-file-and-generate-lrt-file" class="section level2" number="9.5">
<h2><span class="header-section-number">9.5</span> take Beagle file and generate lrt file</h2>
<p>To do the test of genome wide association we need to take our Beagle file and test whether there is an association with our phenotype (in this case whether a fish has a male or female phenotype). The phenotypes are coded as 0 = Female and 1 = Male in the <code>phenobin</code> file.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb7-1"><a href="week-8-genome-wide-association-study-gwas.html#cb7-1" aria-hidden="true" tabindex="-1"></a>$HOME/angsd/angsd -doMaf 4 -beagle salmon_chr2_19ind.BEAGLE.PL.gz -yBin phenobin -doAsso 2 -fai Salmon.fai</span></code></pre></div>
<p>This will generate several output files labeled <code>angsdput</code>. We’ll use the file with the lrt0 extension to plot our manattan plot.</p>
</div>
<div id="take-lrt-file-and-make-a-manhattan-plot" class="section level2" number="9.6">
<h2><span class="header-section-number">9.6</span> take lrt file and make a manhattan plot</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="week-8-genome-wide-association-study-gwas.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#in R</span></span>
<span id="cb8-2"><a href="week-8-genome-wide-association-study-gwas.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co">#set working directory</span></span>
<span id="cb8-3"><a href="week-8-genome-wide-association-study-gwas.html#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="week-8-genome-wide-association-study-gwas.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/home/margeno/MarineGenomicsData/Week8&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="week-8-genome-wide-association-study-gwas.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data</span></span>
<span id="cb9-2"><a href="week-8-genome-wide-association-study-gwas.html#cb9-2" aria-hidden="true" tabindex="-1"></a>lrt<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="fu">gzfile</span>(<span class="st">&quot;angsdput.lrt0.gz&quot;</span>), <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb9-3"><a href="week-8-genome-wide-association-study-gwas.html#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="week-8-genome-wide-association-study-gwas.html#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#look at it</span></span>
<span id="cb9-5"><a href="week-8-genome-wide-association-study-gwas.html#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(lrt)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    224460 obs. of  8 variables:
##  $ Chromosome   : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Position     : int  13 59 74 153 548 550 613 1182 1361 1533 ...
##  $ Major        : Factor w/ 4 levels &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;: 3 3 3 2 2 2 2 2 2 2 ...
##  $ Minor        : Factor w/ 4 levels &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;: 4 1 1 3 4 4 1 4 1 1 ...
##  $ Frequency    : num  0.237 0.5 0.316 0.184 0.316 ...
##  $ N            : int  19 19 19 19 19 19 19 19 19 19 ...
##  $ LRT          : num  -999 -999 -999 -999 -999 -999 -999 -999 -999 -999 ...
##  $ high_WT.HE.HO: Factor w/ 803 levels &quot;0/0/19&quot;,&quot;0/1/18&quot;,..: 125 67 679 198 679 679 372 327 358 372 ...</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="week-8-genome-wide-association-study-gwas.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#we have a few LRT values that are -999, we should remove them. How many do we have?</span></span>
<span id="cb11-2"><a href="week-8-genome-wide-association-study-gwas.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>))</span></code></pre></div>
<pre><code>## [1] 210597</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="week-8-genome-wide-association-study-gwas.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#210597 #most are filtered out</span></span>
<span id="cb13-2"><a href="week-8-genome-wide-association-study-gwas.html#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="week-8-genome-wide-association-study-gwas.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(lrt<span class="sc">$</span>LRT)</span></code></pre></div>
<pre><code>## [1] 224460</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="week-8-genome-wide-association-study-gwas.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#224460</span></span>
<span id="cb15-2"><a href="week-8-genome-wide-association-study-gwas.html#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="week-8-genome-wide-association-study-gwas.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(lrt<span class="sc">$</span>LRT)<span class="sc">-</span><span class="fu">length</span>(<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>))</span></code></pre></div>
<pre><code>## [1] 13863</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="week-8-genome-wide-association-study-gwas.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 13863</span></span>
<span id="cb17-2"><a href="week-8-genome-wide-association-study-gwas.html#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="week-8-genome-wide-association-study-gwas.html#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="week-8-genome-wide-association-study-gwas.html#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co">#remove the values that are not -999 and that are negative</span></span>
<span id="cb17-5"><a href="week-8-genome-wide-association-study-gwas.html#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="week-8-genome-wide-association-study-gwas.html#cb17-6" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="ot">&lt;-</span>lrt[<span class="sc">-</span><span class="fu">c</span>(<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>),<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">&lt;=</span> <span class="dv">0</span>)),]</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="week-8-genome-wide-association-study-gwas.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(lrt_filt<span class="sc">$</span>LRT, <span class="at">breaks=</span><span class="dv">50</span>)</span></code></pre></div>
<p><img src="Test_files/figure-html/8-1-1.png" width="672" /></p>
<p>Everything looks good to proceed to making our manhattan plot.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="week-8-genome-wide-association-study-gwas.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span> (qqman)</span></code></pre></div>
<p>the function manhattan requires each SNP to have it’s own “name.” Let’s make a vector for rownumbers that start with the letter r. We also need to convert our LRT values to pvalues. We’ll use the command <code>dchisq</code> to get pvalues.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="week-8-genome-wide-association-study-gwas.html#cb20-1" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="sc">$</span>SNP<span class="ot">&lt;-</span><span class="fu">paste</span>(<span class="st">&quot;r&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(lrt_filt<span class="sc">$</span>Chromosome), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb20-2"><a href="week-8-genome-wide-association-study-gwas.html#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="week-8-genome-wide-association-study-gwas.html#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co">#we also need to make sure we don&#39;t have any tricky values like those below</span></span>
<span id="cb20-4"><a href="week-8-genome-wide-association-study-gwas.html#cb20-4" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="ot">&lt;-</span>lrt_filt[<span class="sc">-</span><span class="fu">c</span>(<span class="fu">which</span>(lrt_filt<span class="sc">$</span>pvalue <span class="sc">==</span> <span class="st">&quot;NaN&quot;</span> ),</span>
<span id="cb20-5"><a href="week-8-genome-wide-association-study-gwas.html#cb20-5" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">which</span>(lrt_filt<span class="sc">$</span>pvalue <span class="sc">==</span> <span class="st">&quot;Inf&quot;</span>),</span>
<span id="cb20-6"><a href="week-8-genome-wide-association-study-gwas.html#cb20-6" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">which</span>(lrt_filt<span class="sc">$</span>LRT <span class="sc">==</span> <span class="st">&quot;Inf&quot;</span>)),]</span>
<span id="cb20-7"><a href="week-8-genome-wide-association-study-gwas.html#cb20-7" aria-hidden="true" tabindex="-1"></a>                      </span>
<span id="cb20-8"><a href="week-8-genome-wide-association-study-gwas.html#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co">#get pvalues</span></span>
<span id="cb20-9"><a href="week-8-genome-wide-association-study-gwas.html#cb20-9" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="sc">$</span>pvalue<span class="ot">&lt;-</span><span class="fu">dchisq</span>(lrt_filt<span class="sc">$</span>LRT, <span class="at">df=</span><span class="dv">1</span>)</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="week-8-genome-wide-association-study-gwas.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(lrt_filt, <span class="at">chr=</span><span class="st">&quot;Chromosome&quot;</span>, <span class="at">bp=</span><span class="st">&quot;Position&quot;</span>, <span class="at">p=</span><span class="st">&quot;pvalue&quot;</span>)</span></code></pre></div>
<p><img src="Test_files/figure-html/8-2-1.png" width="672" /></p>
<p>Let’s look at a qq-plot of our pvalues to check the model fit</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="week-8-genome-wide-association-study-gwas.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqnorm</span>(lrt_filt<span class="sc">$</span>pvalue)</span></code></pre></div>
<p><img src="Test_files/figure-html/8-3-1.png" width="672" /></p>
<p>This looks a bit weird, we would expect it to be mostly a straight line with some deviations at the upper right. If we were moving forward with this analyses we’d want to do more filtering of our data.</p>
<p>We can highlight the values that are exceed a threshold. There are several ways to determine a threshold, but one is to make a vector of random phenotypes and re-run our association test. We can then set the highest LRT value from the random phenotype test as our upper limit for our plot with the actual phenotypes.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="week-8-genome-wide-association-study-gwas.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">#make a vector with 19 1&#39;s and 0&#39;s</span></span>
<span id="cb23-2"><a href="week-8-genome-wide-association-study-gwas.html#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="week-8-genome-wide-association-study-gwas.html#cb23-3" aria-hidden="true" tabindex="-1"></a>x<span class="ot">&lt;-</span><span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>), <span class="dv">19</span>, <span class="at">replace=</span>T)</span>
<span id="cb23-4"><a href="week-8-genome-wide-association-study-gwas.html#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="week-8-genome-wide-association-study-gwas.html#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co">#write this to our week 8 directory</span></span>
<span id="cb23-6"><a href="week-8-genome-wide-association-study-gwas.html#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="week-8-genome-wide-association-study-gwas.html#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(x, <span class="st">&quot;rando_pheno&quot;</span>, <span class="at">row.names =</span> F)</span></code></pre></div>
<p>And now use that phenotype file to run our association test again, making sure to specify a different output file.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb24-1"><a href="week-8-genome-wide-association-study-gwas.html#cb24-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-2"><a href="week-8-genome-wide-association-study-gwas.html#cb24-2" aria-hidden="true" tabindex="-1"></a>$HOME/angsd/angsd -doMaf 4 -beagle salmon_chr2_19ind.BEAGLE.PL.gz -yBin rando_pheno -doAsso 2 -fai Salmon.fai -out randotest</span></code></pre></div>
<p>And rerun the code in R to see what our maximum LRT values are in this random phenotype test.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="week-8-genome-wide-association-study-gwas.html#cb25-1" aria-hidden="true" tabindex="-1"></a>lrt_rando<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="fu">gzfile</span>(<span class="st">&quot;randotest.lrt0.gz&quot;</span>), <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb25-2"><a href="week-8-genome-wide-association-study-gwas.html#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="week-8-genome-wide-association-study-gwas.html#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co">#we need to remove those -999 values again</span></span>
<span id="cb25-4"><a href="week-8-genome-wide-association-study-gwas.html#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="week-8-genome-wide-association-study-gwas.html#cb25-5" aria-hidden="true" tabindex="-1"></a>rando_filt<span class="ot">&lt;-</span>lrt_rando[<span class="sc">-</span><span class="fu">c</span>(<span class="fu">which</span>(lrt_rando<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>),<span class="fu">which</span>(lrt_rando<span class="sc">$</span>LRT <span class="sc">&lt;=</span> <span class="dv">0</span>)),]</span>
<span id="cb25-6"><a href="week-8-genome-wide-association-study-gwas.html#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="week-8-genome-wide-association-study-gwas.html#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rando_filt<span class="sc">$</span>LRT, <span class="at">na.rm=</span>T)</span></code></pre></div>
<pre><code>##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max.     NA&#39;s 
## 0.000001 0.214816 0.669551      Inf 2.029913      Inf        1</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="week-8-genome-wide-association-study-gwas.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">#we have some Inf values we also need to remove, let&#39;s add those to our filtering line above</span></span>
<span id="cb27-2"><a href="week-8-genome-wide-association-study-gwas.html#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="week-8-genome-wide-association-study-gwas.html#cb27-3" aria-hidden="true" tabindex="-1"></a>rando_filt<span class="ot">&lt;-</span>lrt_rando[<span class="sc">-</span><span class="fu">c</span>(<span class="fu">which</span>(lrt_rando<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>),<span class="fu">which</span>(lrt_rando<span class="sc">$</span>LRT <span class="sc">&lt;=</span> <span class="dv">0</span>), <span class="fu">which</span>(lrt_rando<span class="sc">$</span>LRT <span class="sc">==</span> <span class="cn">Inf</span>)),]</span>
<span id="cb27-4"><a href="week-8-genome-wide-association-study-gwas.html#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="week-8-genome-wide-association-study-gwas.html#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(rando_filt<span class="sc">$</span>LRT, <span class="at">na.rm=</span>T)</span></code></pre></div>
<pre><code>## [1] 12.33287</code></pre>
<p>So we can highlight all of the SNPs that have an LRT greater than 12 in our association test.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="week-8-genome-wide-association-study-gwas.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">#make a list of the candidates</span></span>
<span id="cb29-2"><a href="week-8-genome-wide-association-study-gwas.html#cb29-2" aria-hidden="true" tabindex="-1"></a>candidates<span class="ot">&lt;-</span>lrt_filt[<span class="fu">which</span>(lrt_filt<span class="sc">$</span>LRT <span class="sc">&gt;</span> <span class="dv">36</span>),]<span class="sc">$</span>SNP</span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="week-8-genome-wide-association-study-gwas.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">#refer to that list of candidates with the highlight parameter</span></span>
<span id="cb30-2"><a href="week-8-genome-wide-association-study-gwas.html#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(lrt_filt, <span class="at">chr=</span><span class="st">&quot;Chromosome&quot;</span>, <span class="at">bp=</span><span class="st">&quot;Position&quot;</span>,  <span class="at">p=</span><span class="st">&quot;pvalue&quot;</span>, <span class="at">highlight =</span> candidates)</span></code></pre></div>
<p><img src="Test_files/figure-html/8-4-1.png" width="672" /></p>
<p>Comparing our results to the Kijas et al. 2018 paper, we have a similar pattern of many SNPs across the chromosome showing a relationship with phenotype (sex). Interestingly this paper found that there are three chromosomes that are associated with sex in Atlantic Salmon, but not all chromosomes give a strong signal in all individuals. For example, only three male individuals were found to have a clear association with chromosome 2, and the other males in the study were found to have an association with chromosomes 3 and 6.</p>
<p>These results highlight the fluid nature of sex determination in animals, even those with a genetic basis to sex determination.</p>
<p>For the exercise you’ll take a closer look at chromosome 6, where you’ll try to find the individuals that are male from a PCA plot.</p>
</div>
<div id="exercises-1" class="section level2" number="9.7">
<h2><span class="header-section-number">9.7</span> Exercises</h2>
<blockquote>
<p>1.For this exercise, we want to see which individuals actually show genomic levels of variation at chromosome 6. Using the code that we ran for week 6. Make a PCA plot of the salmon_chr3_19ind.BEAGLE.PL.gz file. Use the function <code>indentify()</code> to find the points that are clustering apart from the other points. Verify that these are males by referring to the table in the Kijas et al. paper <a href="https://www.nature.com/articles/s41598-018-23984-1/tables/1">here</a>. Note their individuals are in the same order as our samples though the names don’t match.</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb31"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb31-1"><a href="week-8-genome-wide-association-study-gwas.html#cb31-1" aria-hidden="true" tabindex="-1"></a>#run pcangsd on our chr6 data</span>
<span id="cb31-2"><a href="week-8-genome-wide-association-study-gwas.html#cb31-2" aria-hidden="true" tabindex="-1"></a>python3 ../../pcangsd/pcangsd.py -beagle salmon_chr6_19ind.BEAGLE.PL.gz -o pca6_out -threads 28</span></code></pre></div>
<p>In R make a PCA plot</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="week-8-genome-wide-association-study-gwas.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data</span></span>
<span id="cb32-2"><a href="week-8-genome-wide-association-study-gwas.html#cb32-2" aria-hidden="true" tabindex="-1"></a>cov<span class="ot">&lt;-</span><span class="fu">as.matrix</span>(<span class="fu">read.table</span>(<span class="st">&quot;pca6_out.cov&quot;</span>))</span>
<span id="cb32-3"><a href="week-8-genome-wide-association-study-gwas.html#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="week-8-genome-wide-association-study-gwas.html#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co">#compute the eigen values</span></span>
<span id="cb32-5"><a href="week-8-genome-wide-association-study-gwas.html#cb32-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-6"><a href="week-8-genome-wide-association-study-gwas.html#cb32-6" aria-hidden="true" tabindex="-1"></a>e<span class="ot">&lt;-</span><span class="fu">eigen</span>(cov)</span>
<span id="cb32-7"><a href="week-8-genome-wide-association-study-gwas.html#cb32-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-8"><a href="week-8-genome-wide-association-study-gwas.html#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="co">#how much variation are we explaining here?</span></span>
<span id="cb32-9"><a href="week-8-genome-wide-association-study-gwas.html#cb32-9" aria-hidden="true" tabindex="-1"></a>e<span class="sc">$</span>values<span class="sc">/</span><span class="fu">sum</span>(e<span class="sc">$</span>values)</span></code></pre></div>
<pre><code>##  [1]  0.1379496011  0.1246126748  0.0969103029  0.0920252715  0.0774570788
##  [6]  0.0685079248  0.0605808079  0.0554445095  0.0493884028  0.0444119246
## [11]  0.0423030396  0.0368221717  0.0329494215  0.0316195242  0.0269138675
## [16]  0.0184940705  0.0149661517 -0.0006450986 -0.0107116466</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="week-8-genome-wide-association-study-gwas.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">#looks like 13.7% is explained by the first axis</span></span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="week-8-genome-wide-association-study-gwas.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#make a plot of the first two axes</span></span>
<span id="cb35-2"><a href="week-8-genome-wide-association-study-gwas.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(e<span class="sc">$</span>vectors[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<p><img src="Test_files/figure-html/8-5-1.png" width="672" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="week-8-genome-wide-association-study-gwas.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co">#use identify to find the points that are clustered apart from the others</span></span>
<span id="cb36-2"><a href="week-8-genome-wide-association-study-gwas.html#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="week-8-genome-wide-association-study-gwas.html#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="co">#identify(e$vectors[,1:2]) This doesn&#39;t work in R studio so we&#39;ll show the plot below</span></span>
<span id="cb36-4"><a href="week-8-genome-wide-association-study-gwas.html#cb36-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-5"><a href="week-8-genome-wide-association-study-gwas.html#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="co">#If you select the points on the left most of the screen, you will find they are rows</span></span>
<span id="cb36-6"><a href="week-8-genome-wide-association-study-gwas.html#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 1, 2, and 12</span></span>
<span id="cb36-7"><a href="week-8-genome-wide-association-study-gwas.html#cb36-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-8"><a href="week-8-genome-wide-association-study-gwas.html#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="co">#If you select the other two points that are leftmost from the remaining points you also get</span></span>
<span id="cb36-9"><a href="week-8-genome-wide-association-study-gwas.html#cb36-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 16 and 19</span></span></code></pre></div>
<p><img src="figs/Rplot_PCACh6.png" /></p>
<p>Looking at the table in the Kilas paper, the Animals in rows 1, 2, and 12 are Male, but it’s rows 17 and 18 that are next not 16 and 19, so we may have somethings out of order in our list.</p>
</p>
</details>
<p> </p>
<blockquote>
<ol start="2" style="list-style-type: decimal">
<li>Make a manhattan plot for chromosome 6. You will need to make a new phenotype file that has 1’s for all the males identified in problem 1. Alterntatively, you can make a phenotype file from table 1 in Kilas et al. 2018, setting the animals that were found to be male on chr 6 to 1 and all other animals to 0.</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb37"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb37-1"><a href="week-8-genome-wide-association-study-gwas.html#cb37-1" aria-hidden="true" tabindex="-1"></a>$ gzip geno_like_filt_allIND.beagle</span></code></pre></div>
<p>and now we can run our pcangsd code</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="week-8-genome-wide-association-study-gwas.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co">#make a phenotype file, note you can also do this in nano or vim in the terminal. Whichever makes sense for you.</span></span>
<span id="cb38-2"><a href="week-8-genome-wide-association-study-gwas.html#cb38-2" aria-hidden="true" tabindex="-1"></a>pheno_chr6<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb38-3"><a href="week-8-genome-wide-association-study-gwas.html#cb38-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-4"><a href="week-8-genome-wide-association-study-gwas.html#cb38-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-5"><a href="week-8-genome-wide-association-study-gwas.html#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(pheno_chr6, <span class="at">file =</span> <span class="st">&quot;phenobin_ch6&quot;</span> )</span>
<span id="cb38-6"><a href="week-8-genome-wide-association-study-gwas.html#cb38-6" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb38-7"><a href="week-8-genome-wide-association-study-gwas.html#cb38-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-8"><a href="week-8-genome-wide-association-study-gwas.html#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="co">#then in the terminal run our command in angsd again using the chr6 BEAGLE file and the phenotype file we just made</span></span>
<span id="cb38-9"><a href="week-8-genome-wide-association-study-gwas.html#cb38-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-10"><a href="week-8-genome-wide-association-study-gwas.html#cb38-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-11"><a href="week-8-genome-wide-association-study-gwas.html#cb38-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $HOME/angsd/angsd -doMaf 4 -beagle salmon_chr6_19ind.BEAGLE.PL.gz -yBin phenobin_ch6 -doAsso 2 -fai Salmon.fai -out chr6_out                 </span></span>
<span id="cb38-12"><a href="week-8-genome-wide-association-study-gwas.html#cb38-12" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb38-13"><a href="week-8-genome-wide-association-study-gwas.html#cb38-13" aria-hidden="true" tabindex="-1"></a><span class="co">#and then we can remake our manhattan plot by copying and pasting the code above</span></span>
<span id="cb38-14"><a href="week-8-genome-wide-association-study-gwas.html#cb38-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-15"><a href="week-8-genome-wide-association-study-gwas.html#cb38-15" aria-hidden="true" tabindex="-1"></a>lrt<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="fu">gzfile</span>(<span class="st">&quot;chr6_out.lrt0.gz&quot;</span>), <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb38-16"><a href="week-8-genome-wide-association-study-gwas.html#cb38-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-17"><a href="week-8-genome-wide-association-study-gwas.html#cb38-17" aria-hidden="true" tabindex="-1"></a><span class="co">#look at it</span></span>
<span id="cb38-18"><a href="week-8-genome-wide-association-study-gwas.html#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(lrt)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    268080 obs. of  8 variables:
##  $ Chromosome   : int  6 6 6 6 6 6 6 6 6 6 ...
##  $ Position     : int  57 1115 1245 4093 4661 4694 6088 6664 7795 7818 ...
##  $ Major        : Factor w/ 4 levels &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;: 2 2 2 4 3 3 2 2 1 1 ...
##  $ Minor        : Factor w/ 4 levels &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;: 3 4 3 3 1 1 4 4 2 4 ...
##  $ Frequency    : num  0.432 0.842 0.289 0.184 0.5 ...
##  $ N            : int  19 19 19 19 19 19 19 19 19 19 ...
##  $ LRT          : num  0.755 -999 -999 -999 1.062 ...
##  $ high_WT.HE.HO: Factor w/ 880 levels &quot;0/0/19&quot;,&quot;0/1/18&quot;,..: 394 84 783 205 119 195 430 189 417 401 ...</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="week-8-genome-wide-association-study-gwas.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">#remove the values that are not -999 and that are negative</span></span>
<span id="cb40-2"><a href="week-8-genome-wide-association-study-gwas.html#cb40-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-3"><a href="week-8-genome-wide-association-study-gwas.html#cb40-3" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="ot">&lt;-</span>lrt[<span class="sc">-</span><span class="fu">c</span>(<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>),<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">&lt;=</span> <span class="dv">0</span>)),]                 </span>
<span id="cb40-4"><a href="week-8-genome-wide-association-study-gwas.html#cb40-4" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="sc">$</span>SNP<span class="ot">&lt;-</span><span class="fu">paste</span>(<span class="st">&quot;r&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(lrt_filt<span class="sc">$</span>Chromosome), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb40-5"><a href="week-8-genome-wide-association-study-gwas.html#cb40-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-6"><a href="week-8-genome-wide-association-study-gwas.html#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="co">#we also need to make sure we don&#39;t have any tricky values like those below</span></span>
<span id="cb40-7"><a href="week-8-genome-wide-association-study-gwas.html#cb40-7" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="ot">&lt;-</span>lrt_filt[<span class="sc">-</span><span class="fu">c</span>(<span class="fu">which</span>(lrt_filt<span class="sc">$</span>pvalue <span class="sc">==</span> <span class="st">&quot;NaN&quot;</span> ),</span>
<span id="cb40-8"><a href="week-8-genome-wide-association-study-gwas.html#cb40-8" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">which</span>(lrt_filt<span class="sc">$</span>pvalue <span class="sc">==</span> <span class="st">&quot;Inf&quot;</span>),</span>
<span id="cb40-9"><a href="week-8-genome-wide-association-study-gwas.html#cb40-9" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">which</span>(lrt_filt<span class="sc">$</span>LRT <span class="sc">==</span> <span class="st">&quot;Inf&quot;</span>)),]</span>
<span id="cb40-10"><a href="week-8-genome-wide-association-study-gwas.html#cb40-10" aria-hidden="true" tabindex="-1"></a>                      </span>
<span id="cb40-11"><a href="week-8-genome-wide-association-study-gwas.html#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co">#get pvalues</span></span>
<span id="cb40-12"><a href="week-8-genome-wide-association-study-gwas.html#cb40-12" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="sc">$</span>pvalue<span class="ot">&lt;-</span><span class="fu">dchisq</span>(lrt_filt<span class="sc">$</span>LRT, <span class="at">df=</span><span class="dv">1</span>)</span>
<span id="cb40-13"><a href="week-8-genome-wide-association-study-gwas.html#cb40-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-14"><a href="week-8-genome-wide-association-study-gwas.html#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="co">#make our plot</span></span>
<span id="cb40-15"><a href="week-8-genome-wide-association-study-gwas.html#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(lrt_filt, <span class="at">chr=</span><span class="st">&quot;Chromosome&quot;</span>, <span class="at">bp=</span><span class="st">&quot;Position&quot;</span>, <span class="at">p=</span><span class="st">&quot;pvalue&quot;</span>)</span></code></pre></div>
<p><img src="Test_files/figure-html/8-6-1.png" width="672" /></p>
</p>
</details>
</div>
<div id="creature-of-the-week-7" class="section level2" number="9.8">
<h2><span class="header-section-number">9.8</span> Creature of the Week!</h2>
<div class="figure">
<img src="figs/creatures/mantisshrimp.png" alt="" />
<p class="caption">The lovely Mantis Shrimp (image credit: Dorothea OLDANI,Unsplash)</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="week-7-fst-and-outlier-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="week-9-population-structure-using-ngsadmix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Test.pdf", "Test.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
