<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>11 Week 9- Population Structure using NGSadmix | Test.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="11 Week 9- Population Structure using NGSadmix | Test.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="11 Week 9- Population Structure using NGSadmix | Test.knit" />
  
  
  

<meta name="author" content="" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="week-8-genome-wide-association-study-gwas.html"/>
<link rel="next" href="week-10--genomics-is-a-blast.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Marine Genomics 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to Marine Genomics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#cloud-computing-versus-individual-laptops"><i class="fa fa-check"></i><b>1.1</b> Cloud computing versus individual laptops</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#meet-our-instructors"><i class="fa fa-check"></i><b>1.2</b> Meet our instructors!</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.3</b> Schedule</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="week-1--welcome.html"><a href="week-1--welcome.html"><i class="fa fa-check"></i><b>2</b> Week 1- Welcome!</a>
<ul>
<li class="chapter" data-level="2.1" data-path="week-1--welcome.html"><a href="week-1--welcome.html#introduction-to-shell-computing-via-the-data-carpentry-tutorial"><i class="fa fa-check"></i><b>2.1</b> Introduction to shell computing via the data carpentry tutorial</a></li>
<li class="chapter" data-level="2.2" data-path="week-1--welcome.html"><a href="week-1--welcome.html#how-to-access-the-shell"><i class="fa fa-check"></i><b>2.2</b> How to access the shell</a></li>
<li class="chapter" data-level="2.3" data-path="week-1--welcome.html"><a href="week-1--welcome.html#week-1-objectives"><i class="fa fa-check"></i><b>2.3</b> Week 1 Objectives</a></li>
<li class="chapter" data-level="2.4" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-your-file-system"><i class="fa fa-check"></i><b>2.4</b> Navigating your file system</a></li>
<li class="chapter" data-level="2.5" data-path="week-1--welcome.html"><a href="week-1--welcome.html#shortcut-tab-completion"><i class="fa fa-check"></i><b>2.5</b> Shortcut: Tab Completion</a></li>
<li class="chapter" data-level="2.6" data-path="week-1--welcome.html"><a href="week-1--welcome.html#summary-key-points"><i class="fa fa-check"></i><b>2.6</b> Summary &amp; Key Points</a></li>
<li class="chapter" data-level="2.7" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-files-and-directories"><i class="fa fa-check"></i><b>2.7</b> Navigating Files and Directories</a></li>
<li class="chapter" data-level="2.8" data-path="week-1--welcome.html"><a href="week-1--welcome.html#moving-around-the-file-system"><i class="fa fa-check"></i><b>2.8</b> Moving around the file system</a></li>
<li class="chapter" data-level="2.9" data-path="week-1--welcome.html"><a href="week-1--welcome.html#examining-the-contents-of-other-directories"><i class="fa fa-check"></i><b>2.9</b> Examining the contents of other directories</a></li>
<li class="chapter" data-level="2.10" data-path="week-1--welcome.html"><a href="week-1--welcome.html#full-vs-relative-paths"><i class="fa fa-check"></i><b>2.10</b> Full vs Relative Paths</a></li>
<li class="chapter" data-level="2.11" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigational-shortcuts"><i class="fa fa-check"></i><b>2.11</b> Navigational shortcuts</a></li>
<li class="chapter" data-level="2.12" data-path="week-1--welcome.html"><a href="week-1--welcome.html#key-points"><i class="fa fa-check"></i><b>2.12</b> Key Points</a></li>
<li class="chapter" data-level="2.13" data-path="week-1--welcome.html"><a href="week-1--welcome.html#creature-of-the-week"><i class="fa fa-check"></i><b>2.13</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html"><i class="fa fa-check"></i><b>3</b> Week 2-Working With Files</a></li>
<li class="chapter" data-level="4" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><i class="fa fa-check"></i><b>4</b> Before we start, make sure to clone or update the github folder MarineGenomics in the user directory</a>
<ul>
<li class="chapter" data-level="4.1" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#our-data-set-fastq-files"><i class="fa fa-check"></i><b>4.1</b> Our data set: FASTQ files</a></li>
<li class="chapter" data-level="4.2" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#wildcards"><i class="fa fa-check"></i><b>4.2</b> Wildcards</a></li>
<li class="chapter" data-level="4.3" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#command-history"><i class="fa fa-check"></i><b>4.3</b> Command History</a></li>
<li class="chapter" data-level="4.4" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#examining-files"><i class="fa fa-check"></i><b>4.4</b> Examining Files</a></li>
<li class="chapter" data-level="4.5" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#details-on-the-fastq-format"><i class="fa fa-check"></i><b>4.5</b> Details on the FASTQ format</a></li>
<li class="chapter" data-level="4.6" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#creating-moving-copying-and-removing"><i class="fa fa-check"></i><b>4.6</b> Creating, moving, copying, and removing</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#copying-files"><i class="fa fa-check"></i><b>4.6.1</b> Copying Files</a></li>
<li class="chapter" data-level="4.6.2" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#creating-directories"><i class="fa fa-check"></i><b>4.6.2</b> Creating Directories</a></li>
<li class="chapter" data-level="4.6.3" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#moving-renaming"><i class="fa fa-check"></i><b>4.6.3</b> Moving / Renaming</a></li>
<li class="chapter" data-level="4.6.4" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#file-permissions"><i class="fa fa-check"></i><b>4.6.4</b> File Permissions</a></li>
<li class="chapter" data-level="4.6.5" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#removing"><i class="fa fa-check"></i><b>4.6.5</b> Removing</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#redirections"><i class="fa fa-check"></i><b>4.7</b> Redirections</a></li>
<li class="chapter" data-level="4.8" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#searching-files"><i class="fa fa-check"></i><b>4.8</b> Searching files</a></li>
<li class="chapter" data-level="4.9" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#redirecting-output"><i class="fa fa-check"></i><b>4.9</b> Redirecting output</a></li>
<li class="chapter" data-level="4.10" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#writing-for-loops"><i class="fa fa-check"></i><b>4.10</b> Writing for loops</a></li>
<li class="chapter" data-level="4.11" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#using-basename-in-for-loops"><i class="fa fa-check"></i><b>4.11</b> Using Basename in for loops</a></li>
<li class="chapter" data-level="4.12" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#writing-scripts-and-working-with-data"><i class="fa fa-check"></i><b>4.12</b> Writing Scripts and Working with Data</a></li>
<li class="chapter" data-level="4.13" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#writing-files"><i class="fa fa-check"></i><b>4.13</b> Writing files</a></li>
<li class="chapter" data-level="4.14" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#writing-scripts"><i class="fa fa-check"></i><b>4.14</b> Writing scripts</a></li>
<li class="chapter" data-level="4.15" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#moving-and-downloading-data"><i class="fa fa-check"></i><b>4.15</b> Moving and Downloading Data</a>
<ul>
<li class="chapter" data-level="4.15.1" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#getting-data-from-the-cloud"><i class="fa fa-check"></i><b>4.15.1</b> Getting data from the cloud</a></li>
<li class="chapter" data-level="4.15.2" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix"><i class="fa fa-check"></i><b>4.15.2</b> Uploading and Downloading Data to your Virtual Machine with scp - UNIX</a></li>
</ul></li>
<li class="chapter" data-level="4.16" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#awk"><i class="fa fa-check"></i><b>4.16</b> Awk</a></li>
<li class="chapter" data-level="4.17" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#creature-of-the-week-1"><i class="fa fa-check"></i><b>4.17</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html"><i class="fa fa-check"></i><b>5</b> Week 3- What is a Genetic Variant?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#to-get-started-lets-dowload-the-data-and-install-a-few-programs"><i class="fa fa-check"></i><b>5.1</b> To get started lets dowload the data and install a few programs</a></li>
<li class="chapter" data-level="5.2" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#raw-read-quality-control"><i class="fa fa-check"></i><b>5.2</b> Raw read quality control</a></li>
<li class="chapter" data-level="5.3" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#trimming-to-remove-adapters"><i class="fa fa-check"></i><b>5.3</b> Trimming to remove adapters</a></li>
<li class="chapter" data-level="5.4" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#building-an-index-of-our-genome"><i class="fa fa-check"></i><b>5.4</b> Building an index of our genome</a></li>
<li class="chapter" data-level="5.5" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#map-reads-to-the-genome"><i class="fa fa-check"></i><b>5.5</b> Map reads to the genome</a></li>
<li class="chapter" data-level="5.6" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#sam-to-bam-file-conversion"><i class="fa fa-check"></i><b>5.6</b> sam to bam file conversion</a></li>
<li class="chapter" data-level="5.7" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#genotype-likelihoods"><i class="fa fa-check"></i><b>5.7</b> Genotype likelihoods</a></li>
<li class="chapter" data-level="5.8" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#creature-of-the-week-2"><i class="fa fa-check"></i><b>5.8</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html"><i class="fa fa-check"></i><b>6</b> Week 4- The R environment</a>
<ul>
<li class="chapter" data-level="6.1" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#lesson-1-orientation-to-r"><i class="fa fa-check"></i><b>6.1</b> Lesson 1: Orientation to R</a></li>
<li class="chapter" data-level="6.2" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#manipulating-a-vector-object"><i class="fa fa-check"></i><b>6.2</b> Manipulating a vector object</a></li>
<li class="chapter" data-level="6.3" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#operations-act-on-each-element-of-a-vector"><i class="fa fa-check"></i><b>6.3</b> Operations act on each element of a vector:</a></li>
<li class="chapter" data-level="6.4" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#operations-can-also-work-with-two-vectors"><i class="fa fa-check"></i><b>6.4</b> Operations can also work with two vectors:</a></li>
<li class="chapter" data-level="6.5" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#a-few-tips-below-for-working-with-objects"><i class="fa fa-check"></i><b>6.5</b> A few tips below for working with objects:</a></li>
<li class="chapter" data-level="6.6" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#practice-r-operations"><i class="fa fa-check"></i><b>6.6</b> Practice R Operations</a></li>
<li class="chapter" data-level="6.7" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#characterizing-a-dataframe"><i class="fa fa-check"></i><b>6.7</b> 1.2 Characterizing a dataframe</a></li>
<li class="chapter" data-level="6.8" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#how-to-access-parts-of-the-data"><i class="fa fa-check"></i><b>6.8</b> How to access parts of the data:</a></li>
<li class="chapter" data-level="6.9" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#data-manipulation"><i class="fa fa-check"></i><b>6.9</b> Data Manipulation</a></li>
<li class="chapter" data-level="6.10" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#practice-exploring-a-dataframe"><i class="fa fa-check"></i><b>6.10</b> Practice exploring a dataframe</a></li>
<li class="chapter" data-level="6.11" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#subsetting-datasets-logicals"><i class="fa fa-check"></i><b>6.11</b> 1.3 Subsetting datasets &amp; logicals</a></li>
<li class="chapter" data-level="6.12" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#practice-subsetting-datasetslogicals"><i class="fa fa-check"></i><b>6.12</b> Practice Subsetting datasets/logicals</a></li>
<li class="chapter" data-level="6.13" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#creature-of-the-week-3"><i class="fa fa-check"></i><b>6.13</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html"><i class="fa fa-check"></i><b>7</b> Week 5- R Continued</a>
<ul>
<li class="chapter" data-level="7.1" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#plotting"><i class="fa fa-check"></i><b>7.1</b> 2.1 Plotting</a></li>
<li class="chapter" data-level="7.2" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#scatterplots"><i class="fa fa-check"></i><b>7.2</b> Scatterplots</a></li>
<li class="chapter" data-level="7.3" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#customizing-your-plot"><i class="fa fa-check"></i><b>7.3</b> Customizing your plot</a></li>
<li class="chapter" data-level="7.4" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#practice-problems-2.1"><i class="fa fa-check"></i><b>7.4</b> Practice Problems 2.1</a></li>
<li class="chapter" data-level="7.5" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#plotting-with-ggplot2"><i class="fa fa-check"></i><b>7.5</b> 2.2 plotting with ggplot2</a></li>
<li class="chapter" data-level="7.6" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#practice-problems-2.2"><i class="fa fa-check"></i><b>7.6</b> Practice Problems 2.2</a></li>
<li class="chapter" data-level="7.7" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#for-loops-and-the-apply-family-of-functions"><i class="fa fa-check"></i><b>7.7</b> For loops and the apply family of functions</a></li>
<li class="chapter" data-level="7.8" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#apply-family"><i class="fa fa-check"></i><b>7.8</b> apply family</a>
<ul>
<li class="chapter" data-level="7.8.1" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#sapply"><i class="fa fa-check"></i><b>7.8.1</b> sapply</a></li>
<li class="chapter" data-level="7.8.2" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#apply"><i class="fa fa-check"></i><b>7.8.2</b> apply</a></li>
<li class="chapter" data-level="7.8.3" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#lapply-list-apply"><i class="fa fa-check"></i><b>7.8.3</b> lapply – “list” apply</a></li>
<li class="chapter" data-level="7.8.4" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#tapply---per-type-apply"><i class="fa fa-check"></i><b>7.8.4</b> tapply - “per Type” apply</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#exercise-2.3-apply-and-tapply"><i class="fa fa-check"></i><b>7.9</b> Exercise 2.3 apply and tapply</a></li>
<li class="chapter" data-level="7.10" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#creature-of-the-week-4"><i class="fa fa-check"></i><b>7.10</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html"><i class="fa fa-check"></i><b>8</b> Week 6- Principle Component Analyses</a>
<ul>
<li class="chapter" data-level="8.1" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#download-the-data"><i class="fa fa-check"></i><b>8.1</b> Download the data</a></li>
<li class="chapter" data-level="8.2" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#installing-programs"><i class="fa fa-check"></i><b>8.2</b> Installing programs</a></li>
<li class="chapter" data-level="8.3" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#run-pcangsd-on-our-data"><i class="fa fa-check"></i><b>8.3</b> Run pcangsd on our data</a></li>
<li class="chapter" data-level="8.4" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#setting-up-a-new-project-in-r"><i class="fa fa-check"></i><b>8.4</b> Setting up a new project in R</a></li>
<li class="chapter" data-level="8.5" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#reading-data-into-r"><i class="fa fa-check"></i><b>8.5</b> Reading data into R</a></li>
<li class="chapter" data-level="8.6" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#exercises"><i class="fa fa-check"></i><b>8.6</b> Exercises</a></li>
<li class="chapter" data-level="8.7" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#creature-of-the-week-5"><i class="fa fa-check"></i><b>8.7</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html"><i class="fa fa-check"></i><b>9</b> Week 7- F<sub>st</sub> and outlier analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#download-the-data-1"><i class="fa fa-check"></i><b>9.1</b> Download the data</a></li>
<li class="chapter" data-level="9.2" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#getting-r-set-up"><i class="fa fa-check"></i><b>9.2</b> Getting R set up</a></li>
<li class="chapter" data-level="9.3" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#finding-outliers-using-pcadapt"><i class="fa fa-check"></i><b>9.3</b> Finding outliers using pcadapt</a></li>
<li class="chapter" data-level="9.4" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#pcadapt-exercises"><i class="fa fa-check"></i><b>9.4</b> pcadapt Exercises</a></li>
<li class="chapter" data-level="9.5" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#using-fst-to-find-outliers"><i class="fa fa-check"></i><b>9.5</b> Using F<sub>st</sub> to find outliers</a></li>
<li class="chapter" data-level="9.6" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#outflank-practice"><i class="fa fa-check"></i><b>9.6</b> OutFLANK Practice</a></li>
<li class="chapter" data-level="9.7" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#creature-of-the-week-6"><i class="fa fa-check"></i><b>9.7</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html"><i class="fa fa-check"></i><b>10</b> Week 8: Genome wide association study (GWAS)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#load-in-the-data"><i class="fa fa-check"></i><b>10.1</b> load in the data</a></li>
<li class="chapter" data-level="10.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-packages-in-r"><i class="fa fa-check"></i><b>10.2</b> install packages in R</a></li>
<li class="chapter" data-level="10.3" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd-and-pcangsd-again"><i class="fa fa-check"></i><b>10.3</b> install angsd and pcangsd (again?)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd"><i class="fa fa-check"></i><b>10.3.1</b> install angsd</a></li>
<li class="chapter" data-level="10.3.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#for-pcangsd"><i class="fa fa-check"></i><b>10.3.2</b> for pcangsd</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#the-data"><i class="fa fa-check"></i><b>10.4</b> The data</a></li>
<li class="chapter" data-level="10.5" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-beagle-file-and-generate-lrt-file"><i class="fa fa-check"></i><b>10.5</b> take Beagle file and generate lrt file</a></li>
<li class="chapter" data-level="10.6" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-lrt-file-and-make-a-manhattan-plot"><i class="fa fa-check"></i><b>10.6</b> take lrt file and make a manhattan plot</a></li>
<li class="chapter" data-level="10.7" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#exercises-1"><i class="fa fa-check"></i><b>10.7</b> Exercises</a></li>
<li class="chapter" data-level="10.8" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#creature-of-the-week-7"><i class="fa fa-check"></i><b>10.8</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html"><i class="fa fa-check"></i><b>11</b> Week 9- Population Structure using NGSadmix</a>
<ul>
<li class="chapter" data-level="11.1" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#download-the-data-2"><i class="fa fa-check"></i><b>11.1</b> Download the data</a></li>
<li class="chapter" data-level="11.2" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#using-ngsadmix"><i class="fa fa-check"></i><b>11.2</b> Using NGSadmix</a></li>
<li class="chapter" data-level="11.3" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#ploting-stucture-in-r"><i class="fa fa-check"></i><b>11.3</b> ploting stucture in R</a></li>
<li class="chapter" data-level="11.4" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#how-do-we-know-which-k-to-pick"><i class="fa fa-check"></i><b>11.4</b> How do we know which K to pick?</a></li>
<li class="chapter" data-level="11.5" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#exercises-2"><i class="fa fa-check"></i><b>11.5</b> Exercises</a></li>
<li class="chapter" data-level="11.6" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#creature-of-the-week-8"><i class="fa fa-check"></i><b>11.6</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html"><i class="fa fa-check"></i><b>12</b> Week 10- Genomics is a BLAST</a>
<ul>
<li class="chapter" data-level="12.1" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#genome-annotation-and-blast-searches"><i class="fa fa-check"></i><b>12.1</b> Genome Annotation and BLAST searches</a></li>
<li class="chapter" data-level="12.2" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#download-the-data-3"><i class="fa fa-check"></i><b>12.2</b> Download the data</a></li>
<li class="chapter" data-level="12.3" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#install-transdecoder"><i class="fa fa-check"></i><b>12.3</b> Install Transdecoder</a></li>
<li class="chapter" data-level="12.4" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#run-transdecoder"><i class="fa fa-check"></i><b>12.4</b> Run Transdecoder</a></li>
<li class="chapter" data-level="12.5" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#run-ncbi-blast"><i class="fa fa-check"></i><b>12.5</b> Run NCBI-BLAST</a></li>
<li class="chapter" data-level="12.6" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#what-did-we-find"><i class="fa fa-check"></i><b>12.6</b> What did we find?</a></li>
<li class="chapter" data-level="12.7" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#the-importance-of-genomic-architecture"><i class="fa fa-check"></i><b>12.7</b> The importance of Genomic Architecture</a></li>
<li class="chapter" data-level="12.8" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#exercises-3"><i class="fa fa-check"></i><b>12.8</b> Exercises</a></li>
<li class="chapter" data-level="12.9" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#creature-of-the-week-9"><i class="fa fa-check"></i><b>12.9</b> Creature of the Week!</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="week-9--population-structure-using-ngsadmix" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">11</span> Week 9- Population Structure using NGSadmix<a href="week-9--population-structure-using-ngsadmix.html#week-9--population-structure-using-ngsadmix" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>For this week we will be exploring another way to document population genetic structure (the first way we covered was PCA in Week 6) via a “Structure” plot implemented in the ANGSD program NGSadmix.</p>
<p>We’ll be using data from the <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.14589">Xuereb et al. paper on P. californicus</a>. This data is a subset of the data published in the paper and is SNP data from 15 individuals from 7 collection sites, in a beagle formated file (the same one we used for our PCA plot in week 6).</p>
<p>The lecture for this week can be found <a href="https://github.com/BayLab/MarineGenomics/blob/4f22a87ad800f98a6e138a166b8e94d016310914/ppt/Week9.pdf">here</a> and describes the basics of a structure plot.</p>
<div id="download-the-data-2" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Download the data<a href="week-9--population-structure-using-ngsadmix.html#download-the-data-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb531"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb531-1"><a href="week-9--population-structure-using-ngsadmix.html#cb531-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-2"><a href="week-9--population-structure-using-ngsadmix.html#cb531-2" aria-hidden="true" tabindex="-1"></a>wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/week9.tar.gz</span>
<span id="cb531-3"><a href="week-9--population-structure-using-ngsadmix.html#cb531-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-4"><a href="week-9--population-structure-using-ngsadmix.html#cb531-4" aria-hidden="true" tabindex="-1"></a>tar -xzvf week9.tar.gz</span>
<span id="cb531-5"><a href="week-9--population-structure-using-ngsadmix.html#cb531-5" aria-hidden="true" tabindex="-1"></a></span></code></pre></div>
<p>You will also need to tar the file within the Week9 folder</p>
<div class="sourceCode" id="cb532"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb532-1"><a href="week-9--population-structure-using-ngsadmix.html#cb532-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-2"><a href="week-9--population-structure-using-ngsadmix.html#cb532-2" aria-hidden="true" tabindex="-1"></a>cd MarineGenomicsData/Week9</span>
<span id="cb532-3"><a href="week-9--population-structure-using-ngsadmix.html#cb532-3" aria-hidden="true" tabindex="-1"></a>tar -xzvf structure_outs.tar.gz</span></code></pre></div>
</div>
<div id="using-ngsadmix" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Using NGSadmix<a href="week-9--population-structure-using-ngsadmix.html#using-ngsadmix" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>NGSadmix is located within the program ANGSD. It’s pretty simple to run and will generate an output file that we can load into R to generate the stucture plot. NGSadmix uses a maximum-likelihood framework to analyse genotype likelihoods, other methods of analyzing population structure including the program STRUCTURE are Bayesian methods and are not modified to be run with genotype likelihoods (but are otherwise widely used and very suitable methods).
<img src="figs/NGS_workflow.png" alt="NGS Workflow" /></p>
<p>Let’s start by running NGSadmix on our beagle file to generate the file for R.</p>
<p>The parameters we need to specify to run NGSadmix are the number of clusters (-K), the minimun minor allele frequency (-minMaf), and the number of threads (-P), and an output file name (-o). You can find the full description of each of these parameters on the NGSadmix <a href="http://www.popgen.dk/software/index.php/NgsAdmix">site</a></p>
<p>The most important thing for us to modify is the number of clusters (-K). This is the paramter we actually want to estimate.</p>
<div class="sourceCode" id="cb533"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb533-1"><a href="week-9--population-structure-using-ngsadmix.html#cb533-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-2"><a href="week-9--population-structure-using-ngsadmix.html#cb533-2" aria-hidden="true" tabindex="-1"></a>../../angsd/misc/NGSadmix -likes genolike_beagle.beagle.gz -K 3 -minMaf 0.05 -seed 1 -o stru_k3 -P 10</span>
<span id="cb533-3"><a href="week-9--population-structure-using-ngsadmix.html#cb533-3" aria-hidden="true" tabindex="-1"></a></span></code></pre></div>
<p>The first part of this code directs from our MarineGenomicsData/Week9 direcotry to the location of the NGSadmix executable located in the <code>misc</code> folder in the angsd program.</p>
<p>We then specify the genotype likelihood file in beagle format with the <code>-like</code> parameter. The remaining parameters are as explained above.</p>
<p>This will output three files which all start with our <code>stru_k5</code> file name. We have a <code>.log</code> file that tells us the loglikihood values and what parameters were run, a <code>.fopt</code> file contains the estimated allele frequencies, and a <code>.qopt</code> file that contains the individuals ancestry proportion (or admixture).</p>
<p>We’ll use the <code>.qopt</code> file to make our plot in R.</p>
</div>
<div id="ploting-stucture-in-r" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> ploting stucture in R<a href="week-9--population-structure-using-ngsadmix.html#ploting-stucture-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Open a new Rscript in RStudio.</p>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="week-9--population-structure-using-ngsadmix.html#cb534-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-2"><a href="week-9--population-structure-using-ngsadmix.html#cb534-2" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data </span></span>
<span id="cb534-3"><a href="week-9--population-structure-using-ngsadmix.html#cb534-3" aria-hidden="true" tabindex="-1"></a>q<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;stru_k3.qopt&quot;</span>)</span>
<span id="cb534-4"><a href="week-9--population-structure-using-ngsadmix.html#cb534-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-5"><a href="week-9--population-structure-using-ngsadmix.html#cb534-5" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the population information file located in the bam.filelist</span></span>
<span id="cb534-6"><a href="week-9--population-structure-using-ngsadmix.html#cb534-6" aria-hidden="true" tabindex="-1"></a>pop<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;bam.filelist&quot;</span>)</span>
<span id="cb534-7"><a href="week-9--population-structure-using-ngsadmix.html#cb534-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-8"><a href="week-9--population-structure-using-ngsadmix.html#cb534-8" aria-hidden="true" tabindex="-1"></a><span class="co">#order it by population</span></span>
<span id="cb534-9"><a href="week-9--population-structure-using-ngsadmix.html#cb534-9" aria-hidden="true" tabindex="-1"></a>ord<span class="ot">&lt;-</span><span class="fu">order</span>(pop<span class="sc">$</span>V2)</span></code></pre></div>
<div class="sourceCode" id="cb535"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb535-1"><a href="week-9--population-structure-using-ngsadmix.html#cb535-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-2"><a href="week-9--population-structure-using-ngsadmix.html#cb535-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(q)[,ord],</span>
<span id="cb535-3"><a href="week-9--population-structure-using-ngsadmix.html#cb535-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,</span>
<span id="cb535-4"><a href="week-9--population-structure-using-ngsadmix.html#cb535-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">names=</span>pop<span class="sc">$</span>V2[ord],</span>
<span id="cb535-5"><a href="week-9--population-structure-using-ngsadmix.html#cb535-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">las=</span><span class="dv">2</span>,</span>
<span id="cb535-6"><a href="week-9--population-structure-using-ngsadmix.html#cb535-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">space=</span><span class="dv">0</span>,</span>
<span id="cb535-7"><a href="week-9--population-structure-using-ngsadmix.html#cb535-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">border=</span><span class="cn">NA</span>,</span>
<span id="cb535-8"><a href="week-9--population-structure-using-ngsadmix.html#cb535-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab=</span><span class="st">&quot;Individuals&quot;</span>,</span>
<span id="cb535-9"><a href="week-9--population-structure-using-ngsadmix.html#cb535-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab=</span><span class="st">&quot;Admixture proportions for K=3&quot;</span>)</span></code></pre></div>
<p><img src="Test_files/figure-html/9-1-1.png" width="768" /></p>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="week-9--population-structure-using-ngsadmix.html#cb536-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb536-2"><a href="week-9--population-structure-using-ngsadmix.html#cb536-2" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data </span></span>
<span id="cb536-3"><a href="week-9--population-structure-using-ngsadmix.html#cb536-3" aria-hidden="true" tabindex="-1"></a>q<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;stru_k7.qopt&quot;</span>)</span>
<span id="cb536-4"><a href="week-9--population-structure-using-ngsadmix.html#cb536-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb536-5"><a href="week-9--population-structure-using-ngsadmix.html#cb536-5" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the population information file located in the bam.filelist</span></span>
<span id="cb536-6"><a href="week-9--population-structure-using-ngsadmix.html#cb536-6" aria-hidden="true" tabindex="-1"></a>pop<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;bam.filelist&quot;</span>)</span>
<span id="cb536-7"><a href="week-9--population-structure-using-ngsadmix.html#cb536-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb536-8"><a href="week-9--population-structure-using-ngsadmix.html#cb536-8" aria-hidden="true" tabindex="-1"></a><span class="co">#order it by population</span></span>
<span id="cb536-9"><a href="week-9--population-structure-using-ngsadmix.html#cb536-9" aria-hidden="true" tabindex="-1"></a>ord<span class="ot">&lt;-</span><span class="fu">order</span>(pop<span class="sc">$</span>V2)</span></code></pre></div>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="week-9--population-structure-using-ngsadmix.html#cb537-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb537-2"><a href="week-9--population-structure-using-ngsadmix.html#cb537-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(q)[,ord],</span>
<span id="cb537-3"><a href="week-9--population-structure-using-ngsadmix.html#cb537-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>,</span>
<span id="cb537-4"><a href="week-9--population-structure-using-ngsadmix.html#cb537-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">names=</span>pop<span class="sc">$</span>V2[ord],</span>
<span id="cb537-5"><a href="week-9--population-structure-using-ngsadmix.html#cb537-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">las=</span><span class="dv">2</span>,</span>
<span id="cb537-6"><a href="week-9--population-structure-using-ngsadmix.html#cb537-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">space=</span><span class="dv">0</span>,</span>
<span id="cb537-7"><a href="week-9--population-structure-using-ngsadmix.html#cb537-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">border=</span><span class="cn">NA</span>,</span>
<span id="cb537-8"><a href="week-9--population-structure-using-ngsadmix.html#cb537-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab=</span><span class="st">&quot;Individuals&quot;</span>,</span>
<span id="cb537-9"><a href="week-9--population-structure-using-ngsadmix.html#cb537-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab=</span><span class="st">&quot;Admixture proportions for K=7&quot;</span>)</span></code></pre></div>
<p><img src="Test_files/figure-html/9-2-1.png" width="768" /></p>
<p>So now we have many more groupings, but they don’t really fall straight across population lines do they?</p>
<p>This shows that we can generate two fairly different looking plots when we change the value of K. How do we tell which one is the “True” value of K? One way is to look at the log-likelihood values from each NGSadmix run.</p>
</div>
<div id="how-do-we-know-which-k-to-pick" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> How do we know which K to pick?<a href="week-9--population-structure-using-ngsadmix.html#how-do-we-know-which-k-to-pick" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In order to get an accurate estimate of the “best” K for our data we need to run NGSadmix on our data for several K values. These values are included in the .tar folder labeled <code>structure_outs</code>. In this example ran NGSadmix for K values 1:7.</p>
<p>We need to read each of these K files into R and pull out the log-likehood values. We’ll then do a test via the Cluster Markov Packager Across K from Evanno which can be seen <a href="http://clumpak.tau.ac.il/index.html">here</a></p>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="week-9--population-structure-using-ngsadmix.html#cb538-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb538-2"><a href="week-9--population-structure-using-ngsadmix.html#cb538-2" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data</span></span>
<span id="cb538-3"><a href="week-9--population-structure-using-ngsadmix.html#cb538-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb538-4"><a href="week-9--population-structure-using-ngsadmix.html#cb538-4" aria-hidden="true" tabindex="-1"></a>data<span class="ot">&lt;-</span><span class="fu">list.files</span>(<span class="st">&quot;structure_outs/&quot;</span>, <span class="at">pattern =</span> <span class="st">&quot;.log&quot;</span>, <span class="at">full.names =</span> T)</span>
<span id="cb538-5"><a href="week-9--population-structure-using-ngsadmix.html#cb538-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb538-6"><a href="week-9--population-structure-using-ngsadmix.html#cb538-6" aria-hidden="true" tabindex="-1"></a><span class="co">#look at data to make sure it only has the log files</span></span>
<span id="cb538-7"><a href="week-9--population-structure-using-ngsadmix.html#cb538-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb538-8"><a href="week-9--population-structure-using-ngsadmix.html#cb538-8" aria-hidden="true" tabindex="-1"></a>data</span>
<span id="cb538-9"><a href="week-9--population-structure-using-ngsadmix.html#cb538-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;structure_outs/stru_k1.2.log&quot; &quot;structure_outs/stru_k1.3.log&quot;</span></span>
<span id="cb538-10"><a href="week-9--population-structure-using-ngsadmix.html#cb538-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] &quot;structure_outs/stru_k1.log&quot;   &quot;structure_outs/stru_k2.2.log&quot;</span></span>
<span id="cb538-11"><a href="week-9--population-structure-using-ngsadmix.html#cb538-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] &quot;structure_outs/stru_k2.3.log&quot; &quot;structure_outs/stru_k2.log&quot;  </span></span>
<span id="cb538-12"><a href="week-9--population-structure-using-ngsadmix.html#cb538-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;structure_outs/stru_k3.1.log&quot; &quot;structure_outs/stru_k3.2.log&quot;</span></span>
<span id="cb538-13"><a href="week-9--population-structure-using-ngsadmix.html#cb538-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;structure_outs/stru_k3.3.log&quot; &quot;structure_outs/stru_k4.2.log&quot;</span></span>
<span id="cb538-14"><a href="week-9--population-structure-using-ngsadmix.html#cb538-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] &quot;structure_outs/stru_k4.3.log&quot; &quot;structure_outs/stru_k4.log&quot;  </span></span>
<span id="cb538-15"><a href="week-9--population-structure-using-ngsadmix.html#cb538-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] &quot;structure_outs/stru_k5.2.log&quot; &quot;structure_outs/stru_k5.3.log&quot;</span></span>
<span id="cb538-16"><a href="week-9--population-structure-using-ngsadmix.html#cb538-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [15] &quot;structure_outs/stru_k5.log&quot;   &quot;structure_outs/stru_k6.2.log&quot;</span></span>
<span id="cb538-17"><a href="week-9--population-structure-using-ngsadmix.html#cb538-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] &quot;structure_outs/stru_k6.3.log&quot; &quot;structure_outs/stru_k6.log&quot;  </span></span>
<span id="cb538-18"><a href="week-9--population-structure-using-ngsadmix.html#cb538-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] &quot;structure_outs/stru_k7.2.log&quot; &quot;structure_outs/stru_k7.3.log&quot;</span></span>
<span id="cb538-19"><a href="week-9--population-structure-using-ngsadmix.html#cb538-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] &quot;structure_outs/stru_k7.log&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb539-1"><a href="week-9--population-structure-using-ngsadmix.html#cb539-1" aria-hidden="true" tabindex="-1"></a><span class="co">#use lapply to read in all our log files at once</span></span>
<span id="cb539-2"><a href="week-9--population-structure-using-ngsadmix.html#cb539-2" aria-hidden="true" tabindex="-1"></a>bigData<span class="ot">&lt;-</span><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="at">FUN =</span> <span class="cf">function</span>(i) <span class="fu">readLines</span>(data[i]))</span>
<span id="cb539-3"><a href="week-9--population-structure-using-ngsadmix.html#cb539-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-4"><a href="week-9--population-structure-using-ngsadmix.html#cb539-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-5"><a href="week-9--population-structure-using-ngsadmix.html#cb539-5" aria-hidden="true" tabindex="-1"></a><span class="co"># find the line that starts with &quot;best like=&quot; or just &quot;b&quot;</span></span>
<span id="cb539-6"><a href="week-9--population-structure-using-ngsadmix.html#cb539-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb539-7"><a href="week-9--population-structure-using-ngsadmix.html#cb539-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-8"><a href="week-9--population-structure-using-ngsadmix.html#cb539-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-9"><a href="week-9--population-structure-using-ngsadmix.html#cb539-9" aria-hidden="true" tabindex="-1"></a><span class="co">#this will pull out the line that starts with &quot;b&quot; from each file and return it as a list</span></span>
<span id="cb539-10"><a href="week-9--population-structure-using-ngsadmix.html#cb539-10" aria-hidden="true" tabindex="-1"></a>foundset<span class="ot">&lt;-</span><span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="at">FUN=</span> <span class="cf">function</span>(x) bigData[[x]][<span class="fu">which</span>(<span class="fu">str_sub</span>(bigData[[x]], <span class="dv">1</span>, <span class="dv">1</span>) <span class="sc">==</span> <span class="st">&#39;b&#39;</span>)])</span>
<span id="cb539-11"><a href="week-9--population-structure-using-ngsadmix.html#cb539-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-12"><a href="week-9--population-structure-using-ngsadmix.html#cb539-12" aria-hidden="true" tabindex="-1"></a>foundset</span>
<span id="cb539-13"><a href="week-9--population-structure-using-ngsadmix.html#cb539-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;best like=-3535959.455618 after 12 iterations&quot; </span></span>
<span id="cb539-14"><a href="week-9--population-structure-using-ngsadmix.html#cb539-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [2] &quot;best like=-3535959.455618 after 12 iterations&quot; </span></span>
<span id="cb539-15"><a href="week-9--population-structure-using-ngsadmix.html#cb539-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] &quot;best like=-3535959.455618 after 12 iterations&quot; </span></span>
<span id="cb539-16"><a href="week-9--population-structure-using-ngsadmix.html#cb539-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [4] &quot;best like=-3308469.603529 after 250 iterations&quot;</span></span>
<span id="cb539-17"><a href="week-9--population-structure-using-ngsadmix.html#cb539-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] &quot;best like=-3312314.435844 after 300 iterations&quot;</span></span>
<span id="cb539-18"><a href="week-9--population-structure-using-ngsadmix.html#cb539-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [6] &quot;best like=-3318928.120535 after 440 iterations&quot;</span></span>
<span id="cb539-19"><a href="week-9--population-structure-using-ngsadmix.html#cb539-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;best like=-3136880.875876 after 250 iterations&quot;</span></span>
<span id="cb539-20"><a href="week-9--population-structure-using-ngsadmix.html#cb539-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [8] &quot;best like=-3144328.694755 after 250 iterations&quot;</span></span>
<span id="cb539-21"><a href="week-9--population-structure-using-ngsadmix.html#cb539-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;best like=-3141386.218416 after 672 iterations&quot;</span></span>
<span id="cb539-22"><a href="week-9--population-structure-using-ngsadmix.html#cb539-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [10] &quot;best like=-3006651.601488 after 210 iterations&quot;</span></span>
<span id="cb539-23"><a href="week-9--population-structure-using-ngsadmix.html#cb539-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] &quot;best like=-3000275.945373 after 184 iterations&quot;</span></span>
<span id="cb539-24"><a href="week-9--population-structure-using-ngsadmix.html#cb539-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [12] &quot;best like=-3000534.312886 after 250 iterations&quot;</span></span>
<span id="cb539-25"><a href="week-9--population-structure-using-ngsadmix.html#cb539-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] &quot;best like=-2892659.793895 after 657 iterations&quot;</span></span>
<span id="cb539-26"><a href="week-9--population-structure-using-ngsadmix.html#cb539-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [14] &quot;best like=-2881306.372706 after 650 iterations&quot;</span></span>
<span id="cb539-27"><a href="week-9--population-structure-using-ngsadmix.html#cb539-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [15] &quot;best like=-2874661.629230 after 600 iterations&quot;</span></span>
<span id="cb539-28"><a href="week-9--population-structure-using-ngsadmix.html#cb539-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [16] &quot;best like=-2771520.855761 after 450 iterations&quot;</span></span>
<span id="cb539-29"><a href="week-9--population-structure-using-ngsadmix.html#cb539-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] &quot;best like=-2776458.073377 after 229 iterations&quot;</span></span>
<span id="cb539-30"><a href="week-9--population-structure-using-ngsadmix.html#cb539-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [18] &quot;best like=-2774537.683445 after 487 iterations&quot;</span></span>
<span id="cb539-31"><a href="week-9--population-structure-using-ngsadmix.html#cb539-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] &quot;best like=-2690599.496764 after 250 iterations&quot;</span></span>
<span id="cb539-32"><a href="week-9--population-structure-using-ngsadmix.html#cb539-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [20] &quot;best like=-2671043.271473 after 300 iterations&quot;</span></span>
<span id="cb539-33"><a href="week-9--population-structure-using-ngsadmix.html#cb539-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] &quot;best like=-2694025.576965 after 233 iterations&quot;</span></span>
<span id="cb539-34"><a href="week-9--population-structure-using-ngsadmix.html#cb539-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-35"><a href="week-9--population-structure-using-ngsadmix.html#cb539-35" aria-hidden="true" tabindex="-1"></a><span class="co">#we&#39;re getting there!</span></span>
<span id="cb539-36"><a href="week-9--population-structure-using-ngsadmix.html#cb539-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-37"><a href="week-9--population-structure-using-ngsadmix.html#cb539-37" aria-hidden="true" tabindex="-1"></a><span class="co">#now we need to pull out the first number in the string, we&#39;ll do this with the function sub</span></span>
<span id="cb539-38"><a href="week-9--population-structure-using-ngsadmix.html#cb539-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-39"><a href="week-9--population-structure-using-ngsadmix.html#cb539-39" aria-hidden="true" tabindex="-1"></a><span class="fu">as.numeric</span>( <span class="fu">sub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">D*(</span><span class="sc">\\</span><span class="st">d+).*&quot;</span>, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">1&quot;</span>, foundset) )</span>
<span id="cb539-40"><a href="week-9--population-structure-using-ngsadmix.html#cb539-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] 3535959 3535959 3535959 3308469 3312314 3318928 3136880 3144328 3141386 3006651</span></span>
<span id="cb539-41"><a href="week-9--population-structure-using-ngsadmix.html#cb539-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] 3000275 3000534 2892659 2881306 2874661 2771520 2776458 2774537 2690599 2671043</span></span>
<span id="cb539-42"><a href="week-9--population-structure-using-ngsadmix.html#cb539-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] 2694025</span></span>
<span id="cb539-43"><a href="week-9--population-structure-using-ngsadmix.html#cb539-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-44"><a href="week-9--population-structure-using-ngsadmix.html#cb539-44" aria-hidden="true" tabindex="-1"></a><span class="co">#now lets store it in a dataframe</span></span>
<span id="cb539-45"><a href="week-9--population-structure-using-ngsadmix.html#cb539-45" aria-hidden="true" tabindex="-1"></a><span class="co">#make a dataframe with an index 1:7, this corresponds to our K values</span></span>
<span id="cb539-46"><a href="week-9--population-structure-using-ngsadmix.html#cb539-46" aria-hidden="true" tabindex="-1"></a>logs<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="at">K =</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>, <span class="at">each=</span><span class="dv">3</span>))</span>
<span id="cb539-47"><a href="week-9--population-structure-using-ngsadmix.html#cb539-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-48"><a href="week-9--population-structure-using-ngsadmix.html#cb539-48" aria-hidden="true" tabindex="-1"></a><span class="co">#add to it our likelihood values</span></span>
<span id="cb539-49"><a href="week-9--population-structure-using-ngsadmix.html#cb539-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-50"><a href="week-9--population-structure-using-ngsadmix.html#cb539-50" aria-hidden="true" tabindex="-1"></a>logs<span class="sc">$</span>like<span class="ot">&lt;-</span><span class="fu">as.vector</span>(<span class="fu">as.numeric</span>( <span class="fu">sub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">D*(</span><span class="sc">\\</span><span class="st">d+).*&quot;</span>, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">1&quot;</span>, foundset) ))</span>
<span id="cb539-51"><a href="week-9--population-structure-using-ngsadmix.html#cb539-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-52"><a href="week-9--population-structure-using-ngsadmix.html#cb539-52" aria-hidden="true" tabindex="-1"></a><span class="co">#and now we can calculate our delta K and probability</span></span>
<span id="cb539-53"><a href="week-9--population-structure-using-ngsadmix.html#cb539-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-54"><a href="week-9--population-structure-using-ngsadmix.html#cb539-54" aria-hidden="true" tabindex="-1"></a><span class="fu">tapply</span>(logs<span class="sc">$</span>like, logs<span class="sc">$</span>K, <span class="at">FUN=</span> <span class="cf">function</span>(x) <span class="fu">mean</span>(<span class="fu">abs</span>(x))<span class="sc">/</span><span class="fu">sd</span>(<span class="fu">abs</span>(x)))</span>
<span id="cb539-55"><a href="week-9--population-structure-using-ngsadmix.html#cb539-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         1         2         3         4         5         6         7 </span></span>
<span id="cb539-56"><a href="week-9--population-structure-using-ngsadmix.html#cb539-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       Inf  626.2926  837.2806  832.0035  316.7629 1114.4881  216.5752</span></span></code></pre></div>
<p>We then use these values to select our K, which will be the one that has the highest value, so in this case 6. Note these values are rather high because our variance is so low. We would normally want to re-run the NGSadmix analyses more than 3 times to generate these values, but this illustrates the process.</p>
</div>
<div id="exercises-2" class="section level2 hasAnchor" number="11.5">
<h2><span class="header-section-number">11.5</span> Exercises<a href="week-9--population-structure-using-ngsadmix.html#exercises-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<h1 id="practice-questions-3" class="hasAnchor">Practice Questions:<a href="week-9--population-structure-using-ngsadmix.html#practice-questions-3" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<ol style="list-style-type: decimal">
<li>Remake the structure plot using K=6 <code>.qopt</code>. You’ll find this file in the <code>structure_outs</code> file.</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: Orchid;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="week-9--population-structure-using-ngsadmix.html#cb540-1" aria-hidden="true" tabindex="-1"></a>q<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;stru_k6.qopt&quot;</span>)</span>
<span id="cb540-2"><a href="week-9--population-structure-using-ngsadmix.html#cb540-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb540-3"><a href="week-9--population-structure-using-ngsadmix.html#cb540-3" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the population information file located in the bam.filelist</span></span>
<span id="cb540-4"><a href="week-9--population-structure-using-ngsadmix.html#cb540-4" aria-hidden="true" tabindex="-1"></a>pop<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;bam.filelist&quot;</span>)</span>
<span id="cb540-5"><a href="week-9--population-structure-using-ngsadmix.html#cb540-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb540-6"><a href="week-9--population-structure-using-ngsadmix.html#cb540-6" aria-hidden="true" tabindex="-1"></a><span class="co">#order it by population</span></span>
<span id="cb540-7"><a href="week-9--population-structure-using-ngsadmix.html#cb540-7" aria-hidden="true" tabindex="-1"></a>ord<span class="ot">&lt;-</span><span class="fu">order</span>(pop<span class="sc">$</span>V2)</span></code></pre></div>
<div class="sourceCode" id="cb541"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb541-1"><a href="week-9--population-structure-using-ngsadmix.html#cb541-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb541-2"><a href="week-9--population-structure-using-ngsadmix.html#cb541-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(q)[,ord],</span>
<span id="cb541-3"><a href="week-9--population-structure-using-ngsadmix.html#cb541-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>,</span>
<span id="cb541-4"><a href="week-9--population-structure-using-ngsadmix.html#cb541-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">names=</span>pop<span class="sc">$</span>V2[ord],</span>
<span id="cb541-5"><a href="week-9--population-structure-using-ngsadmix.html#cb541-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">las=</span><span class="dv">2</span>,</span>
<span id="cb541-6"><a href="week-9--population-structure-using-ngsadmix.html#cb541-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">space=</span><span class="dv">0</span>,</span>
<span id="cb541-7"><a href="week-9--population-structure-using-ngsadmix.html#cb541-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">border=</span><span class="cn">NA</span>,</span>
<span id="cb541-8"><a href="week-9--population-structure-using-ngsadmix.html#cb541-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab=</span><span class="st">&quot;Individuals&quot;</span>,</span>
<span id="cb541-9"><a href="week-9--population-structure-using-ngsadmix.html#cb541-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab=</span><span class="st">&quot;Admixture proportions for K=6&quot;</span>)</span></code></pre></div>
<img src="Test_files/figure-html/9-3-1.png" width="768" />
</p>
</details>
<blockquote>
<ol start="2" style="list-style-type: decimal">
<li>Make a multi-panel plot showing K=2, K=4, and K=6. You can set the parameters of the plot screen to allow three rows and one column using par(mfrow=c(3,1)). For this plot use the region identifier instead of the name (V3 in the bam file so pop$V3)</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: Orchid;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="week-9--population-structure-using-ngsadmix.html#cb542-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb542-2"><a href="week-9--population-structure-using-ngsadmix.html#cb542-2" aria-hidden="true" tabindex="-1"></a><span class="co">#read in all the data </span></span>
<span id="cb542-3"><a href="week-9--population-structure-using-ngsadmix.html#cb542-3" aria-hidden="true" tabindex="-1"></a>q2<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;stru_k2.qopt&quot;</span>)</span>
<span id="cb542-4"><a href="week-9--population-structure-using-ngsadmix.html#cb542-4" aria-hidden="true" tabindex="-1"></a>q4<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;stru_k4.qopt&quot;</span>)</span>
<span id="cb542-5"><a href="week-9--population-structure-using-ngsadmix.html#cb542-5" aria-hidden="true" tabindex="-1"></a>q6<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;stru_k6.qopt&quot;</span>)</span>
<span id="cb542-6"><a href="week-9--population-structure-using-ngsadmix.html#cb542-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb542-7"><a href="week-9--population-structure-using-ngsadmix.html#cb542-7" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the population information file located in the bam.filelist</span></span>
<span id="cb542-8"><a href="week-9--population-structure-using-ngsadmix.html#cb542-8" aria-hidden="true" tabindex="-1"></a>pop<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;bam.filelist&quot;</span>)</span>
<span id="cb542-9"><a href="week-9--population-structure-using-ngsadmix.html#cb542-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb542-10"><a href="week-9--population-structure-using-ngsadmix.html#cb542-10" aria-hidden="true" tabindex="-1"></a><span class="co">#order it by population</span></span>
<span id="cb542-11"><a href="week-9--population-structure-using-ngsadmix.html#cb542-11" aria-hidden="true" tabindex="-1"></a>ord<span class="ot">&lt;-</span><span class="fu">order</span>(pop<span class="sc">$</span>V3)</span>
<span id="cb542-12"><a href="week-9--population-structure-using-ngsadmix.html#cb542-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb542-13"><a href="week-9--population-structure-using-ngsadmix.html#cb542-13" aria-hidden="true" tabindex="-1"></a><span class="co">#set the parameters for our screen to show three rows and one column</span></span>
<span id="cb542-14"><a href="week-9--population-structure-using-ngsadmix.html#cb542-14" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>))</span></code></pre></div>
<div class="sourceCode" id="cb543"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb543-1"><a href="week-9--population-structure-using-ngsadmix.html#cb543-1" aria-hidden="true" tabindex="-1"></a><span class="co">#plot the plots</span></span>
<span id="cb543-2"><a href="week-9--population-structure-using-ngsadmix.html#cb543-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb543-3"><a href="week-9--population-structure-using-ngsadmix.html#cb543-3" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(q2)[,ord],</span>
<span id="cb543-4"><a href="week-9--population-structure-using-ngsadmix.html#cb543-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,</span>
<span id="cb543-5"><a href="week-9--population-structure-using-ngsadmix.html#cb543-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">names=</span>pop<span class="sc">$</span>V3[ord],</span>
<span id="cb543-6"><a href="week-9--population-structure-using-ngsadmix.html#cb543-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">las=</span><span class="dv">2</span>,</span>
<span id="cb543-7"><a href="week-9--population-structure-using-ngsadmix.html#cb543-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">space=</span><span class="dv">0</span>,</span>
<span id="cb543-8"><a href="week-9--population-structure-using-ngsadmix.html#cb543-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">border=</span><span class="cn">NA</span>,</span>
<span id="cb543-9"><a href="week-9--population-structure-using-ngsadmix.html#cb543-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab=</span><span class="st">&quot;Individuals&quot;</span>,</span>
<span id="cb543-10"><a href="week-9--population-structure-using-ngsadmix.html#cb543-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab=</span><span class="st">&quot;Admixture prop K=2&quot;</span>)</span></code></pre></div>
<p><img src="Test_files/figure-html/9-4-1.png" width="768" /></p>
<div class="sourceCode" id="cb544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb544-1"><a href="week-9--population-structure-using-ngsadmix.html#cb544-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb544-2"><a href="week-9--population-structure-using-ngsadmix.html#cb544-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(q4)[,ord],</span>
<span id="cb544-3"><a href="week-9--population-structure-using-ngsadmix.html#cb544-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,</span>
<span id="cb544-4"><a href="week-9--population-structure-using-ngsadmix.html#cb544-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">names=</span>pop<span class="sc">$</span>V3[ord],</span>
<span id="cb544-5"><a href="week-9--population-structure-using-ngsadmix.html#cb544-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">las=</span><span class="dv">2</span>,</span>
<span id="cb544-6"><a href="week-9--population-structure-using-ngsadmix.html#cb544-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">space=</span><span class="dv">0</span>,</span>
<span id="cb544-7"><a href="week-9--population-structure-using-ngsadmix.html#cb544-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">border=</span><span class="cn">NA</span>,</span>
<span id="cb544-8"><a href="week-9--population-structure-using-ngsadmix.html#cb544-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab=</span><span class="st">&quot;Individuals&quot;</span>,</span>
<span id="cb544-9"><a href="week-9--population-structure-using-ngsadmix.html#cb544-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab=</span><span class="st">&quot;Admixture prop K=4&quot;</span>)</span></code></pre></div>
<p><img src="Test_files/figure-html/9-4-2.png" width="768" /></p>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="week-9--population-structure-using-ngsadmix.html#cb545-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb545-2"><a href="week-9--population-structure-using-ngsadmix.html#cb545-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(q6)[,ord],</span>
<span id="cb545-3"><a href="week-9--population-structure-using-ngsadmix.html#cb545-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">col=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>,</span>
<span id="cb545-4"><a href="week-9--population-structure-using-ngsadmix.html#cb545-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">names=</span>pop<span class="sc">$</span>V3[ord],</span>
<span id="cb545-5"><a href="week-9--population-structure-using-ngsadmix.html#cb545-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">las=</span><span class="dv">2</span>,</span>
<span id="cb545-6"><a href="week-9--population-structure-using-ngsadmix.html#cb545-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">space=</span><span class="dv">0</span>,</span>
<span id="cb545-7"><a href="week-9--population-structure-using-ngsadmix.html#cb545-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">border=</span><span class="cn">NA</span>,</span>
<span id="cb545-8"><a href="week-9--population-structure-using-ngsadmix.html#cb545-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab=</span><span class="st">&quot;Individuals&quot;</span>,</span>
<span id="cb545-9"><a href="week-9--population-structure-using-ngsadmix.html#cb545-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab=</span><span class="st">&quot;Admixture prop K=6&quot;</span>)</span></code></pre></div>
<img src="Test_files/figure-html/9-4-3.png" width="768" />
</p>
</details>
</div>
<div id="creature-of-the-week-8" class="section level2 hasAnchor" number="11.6">
<h2><span class="header-section-number">11.6</span> Creature of the Week!<a href="week-9--population-structure-using-ngsadmix.html#creature-of-the-week-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure">
<img src="figs/creatures/bearded_fireworm.jpg" alt="" />
<p class="caption">The Bearded Fireworm (image credit: Philippe Guillaume via Flickr)</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="week-8-genome-wide-association-study-gwas.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="week-10--genomics-is-a-blast.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Test.pdf", "Test.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
