<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Week 6: PCA | Week 6</title>
  <meta name="description" content="7 Week 6: PCA | Week 6" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Week 6: PCA | Week 6" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Week 6: PCA | Week 6" />
  
  
  

<meta name="author" content="Serena Caplins" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="week-5-r-continued.html"/>
<link rel="next" href="week-7-fst-and-outlier-analysis.html"/>
<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Marine Genomics 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to Marine Genomics</a></li>
<li class="chapter" data-level="2" data-path="week-1-welcome.html"><a href="week-1-welcome.html"><i class="fa fa-check"></i><b>2</b> Week 1- Welcome!</a>
<ul>
<li class="chapter" data-level="2.1" data-path="week-1-welcome.html"><a href="week-1-welcome.html#introduction-to-shell-computing-via-the-data-carpentry-tutorial"><i class="fa fa-check"></i><b>2.1</b> Introduction to shell computing via the data carpentry tutorial</a></li>
<li class="chapter" data-level="2.2" data-path="week-1-welcome.html"><a href="week-1-welcome.html#how-to-access-the-shell"><i class="fa fa-check"></i><b>2.2</b> How to access the shell</a></li>
<li class="chapter" data-level="2.3" data-path="week-1-welcome.html"><a href="week-1-welcome.html#navigating-your-file-system"><i class="fa fa-check"></i><b>2.3</b> Navigating your file system</a></li>
<li class="chapter" data-level="2.4" data-path="week-1-welcome.html"><a href="week-1-welcome.html#shortcut-tab-completion"><i class="fa fa-check"></i><b>2.4</b> Shortcut: Tab Completion</a></li>
<li class="chapter" data-level="2.5" data-path="week-1-welcome.html"><a href="week-1-welcome.html#summary-key-points"><i class="fa fa-check"></i><b>2.5</b> Summary &amp; Key Points</a></li>
<li class="chapter" data-level="2.6" data-path="week-1-welcome.html"><a href="week-1-welcome.html#navigating-files-and-directories"><i class="fa fa-check"></i><b>2.6</b> Navigating Files and Directories</a></li>
<li class="chapter" data-level="2.7" data-path="week-1-welcome.html"><a href="week-1-welcome.html#week-1-objectives"><i class="fa fa-check"></i><b>2.7</b> Week 1 Objectives</a></li>
<li class="chapter" data-level="2.8" data-path="week-1-welcome.html"><a href="week-1-welcome.html#moving-around-the-file-system"><i class="fa fa-check"></i><b>2.8</b> Moving around the file system</a></li>
<li class="chapter" data-level="2.9" data-path="week-1-welcome.html"><a href="week-1-welcome.html#examining-the-contents-of-other-directories"><i class="fa fa-check"></i><b>2.9</b> Examining the contents of other directories</a></li>
<li class="chapter" data-level="2.10" data-path="week-1-welcome.html"><a href="week-1-welcome.html#full-vs-relative-paths"><i class="fa fa-check"></i><b>2.10</b> Full vs Relative Paths</a></li>
<li class="chapter" data-level="2.11" data-path="week-1-welcome.html"><a href="week-1-welcome.html#navigational-shortcuts"><i class="fa fa-check"></i><b>2.11</b> Navigational shortcuts</a></li>
<li class="chapter" data-level="2.12" data-path="week-1-welcome.html"><a href="week-1-welcome.html#key-points"><i class="fa fa-check"></i><b>2.12</b> Key Points</a></li>
<li class="chapter" data-level="2.13" data-path="week-1-welcome.html"><a href="week-1-welcome.html#creature-of-the-week"><i class="fa fa-check"></i><b>2.13</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html"><i class="fa fa-check"></i><b>3</b> Week 2-Working With Files</a>
<ul>
<li class="chapter" data-level="3.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#our-data-set-fastq-files"><i class="fa fa-check"></i><b>3.1</b> Our data set: FASTQ files</a></li>
<li class="chapter" data-level="3.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#wildcards"><i class="fa fa-check"></i><b>3.2</b> Wildcards</a></li>
<li class="chapter" data-level="3.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#command-history"><i class="fa fa-check"></i><b>3.3</b> Command History</a></li>
<li class="chapter" data-level="3.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#examining-files"><i class="fa fa-check"></i><b>3.4</b> Examining Files</a></li>
<li class="chapter" data-level="3.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#details-on-the-fastq-format"><i class="fa fa-check"></i><b>3.5</b> Details on the FASTQ format</a></li>
<li class="chapter" data-level="3.6" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-moving-copying-and-removing"><i class="fa fa-check"></i><b>3.6</b> Creating, moving, copying, and removing</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#copying-files"><i class="fa fa-check"></i><b>3.6.1</b> Copying Files</a></li>
<li class="chapter" data-level="3.6.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-directories"><i class="fa fa-check"></i><b>3.6.2</b> Creating Directories</a></li>
<li class="chapter" data-level="3.6.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#moving-renaming"><i class="fa fa-check"></i><b>3.6.3</b> Moving / Renaming</a></li>
<li class="chapter" data-level="3.6.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#file-permissions"><i class="fa fa-check"></i><b>3.6.4</b> File Permissions</a></li>
<li class="chapter" data-level="3.6.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#removing"><i class="fa fa-check"></i><b>3.6.5</b> Removing</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#redirections"><i class="fa fa-check"></i><b>3.7</b> Redirections</a></li>
<li class="chapter" data-level="3.8" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#searching-files"><i class="fa fa-check"></i><b>3.8</b> Searching files</a></li>
<li class="chapter" data-level="3.9" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#redirecting-output"><i class="fa fa-check"></i><b>3.9</b> Redirecting output</a></li>
<li class="chapter" data-level="3.10" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-for-loops"><i class="fa fa-check"></i><b>3.10</b> Writing for loops</a></li>
<li class="chapter" data-level="3.11" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#using-basename-in-for-loops"><i class="fa fa-check"></i><b>3.11</b> Using Basename in for loops</a></li>
<li class="chapter" data-level="3.12" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts-and-working-with-data"><i class="fa fa-check"></i><b>3.12</b> Writing Scripts and Working with Data</a></li>
<li class="chapter" data-level="3.13" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-files"><i class="fa fa-check"></i><b>3.13</b> Writing files</a></li>
<li class="chapter" data-level="3.14" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts"><i class="fa fa-check"></i><b>3.14</b> Writing scripts</a></li>
<li class="chapter" data-level="3.15" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#moving-and-downloading-data"><i class="fa fa-check"></i><b>3.15</b> Moving and Downloading Data</a>
<ul>
<li class="chapter" data-level="3.15.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#getting-data-from-the-cloud"><i class="fa fa-check"></i><b>3.15.1</b> Getting data from the cloud</a></li>
<li class="chapter" data-level="3.15.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix"><i class="fa fa-check"></i><b>3.15.2</b> Uploading and Downloading Data to your Virtual Machine with scp - UNIX</a></li>
</ul></li>
<li class="chapter" data-level="3.16" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#awk"><i class="fa fa-check"></i><b>3.16</b> Awk</a></li>
<li class="chapter" data-level="3.17" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creature-of-the-week-1"><i class="fa fa-check"></i><b>3.17</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html"><i class="fa fa-check"></i><b>4</b> Week 3- What is a Genetic Variant?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#to-get-started-lets-dowload-the-data-and-install-a-few-programs"><i class="fa fa-check"></i><b>4.1</b> To get started lets dowload the data and install a few programs</a></li>
<li class="chapter" data-level="4.2" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#raw-read-quality-control"><i class="fa fa-check"></i><b>4.2</b> Raw read quality control</a></li>
<li class="chapter" data-level="4.3" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#trimming-to-remove-adapters"><i class="fa fa-check"></i><b>4.3</b> Trimming to remove adapters</a></li>
<li class="chapter" data-level="4.4" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#building-an-index-of-our-genome"><i class="fa fa-check"></i><b>4.4</b> Building an index of our genome</a></li>
<li class="chapter" data-level="4.5" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#map-reads-to-the-genome"><i class="fa fa-check"></i><b>4.5</b> Map reads to the genome</a></li>
<li class="chapter" data-level="4.6" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#sam-to-bam-file-conversion"><i class="fa fa-check"></i><b>4.6</b> sam to bam file conversion</a></li>
<li class="chapter" data-level="4.7" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#genotype-likelihoods"><i class="fa fa-check"></i><b>4.7</b> Genotype likelihoods</a></li>
<li class="chapter" data-level="4.8" data-path="week-3-what-is-a-genetic-variant.html"><a href="week-3-what-is-a-genetic-variant.html#creature-of-the-week-2"><i class="fa fa-check"></i><b>4.8</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html"><i class="fa fa-check"></i><b>5</b> Week 4- The R environment</a>
<ul>
<li class="chapter" data-level="5.1" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#lesson-1-orientation-to-r"><i class="fa fa-check"></i><b>5.1</b> Lesson 1: Orientation to R</a></li>
<li class="chapter" data-level="5.2" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#manipulating-a-vector-object"><i class="fa fa-check"></i><b>5.2</b> Manipulating a vector object</a></li>
<li class="chapter" data-level="5.3" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#operations-act-on-each-element-of-a-vector"><i class="fa fa-check"></i><b>5.3</b> Operations act on each element of a vector:</a></li>
<li class="chapter" data-level="5.4" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#operations-can-also-work-with-two-vectors"><i class="fa fa-check"></i><b>5.4</b> Operations can also work with two vectors:</a></li>
<li class="chapter" data-level="5.5" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#a-few-tips-below-for-working-with-objects"><i class="fa fa-check"></i><b>5.5</b> A few tips below for working with objects:</a></li>
<li class="chapter" data-level="5.6" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#practice-r-operations"><i class="fa fa-check"></i><b>5.6</b> Practice R Operations</a></li>
<li class="chapter" data-level="5.7" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#characterizing-a-dataframe"><i class="fa fa-check"></i><b>5.7</b> 1.2 Characterizing a dataframe</a></li>
<li class="chapter" data-level="5.8" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#how-to-access-parts-of-the-data"><i class="fa fa-check"></i><b>5.8</b> How to access parts of the data:</a></li>
<li class="chapter" data-level="5.9" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#data-manipulation"><i class="fa fa-check"></i><b>5.9</b> Data Manipulation</a></li>
<li class="chapter" data-level="5.10" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#practice-exploring-a-dataframe"><i class="fa fa-check"></i><b>5.10</b> Practice exploring a dataframe</a></li>
<li class="chapter" data-level="5.11" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#subsetting-datasets-logicals"><i class="fa fa-check"></i><b>5.11</b> 1.3 Subsetting datasets &amp; logicals</a></li>
<li class="chapter" data-level="5.12" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#practice-subsetting-datasetslogicals"><i class="fa fa-check"></i><b>5.12</b> Practice Subsetting datasets/logicals</a></li>
<li class="chapter" data-level="5.13" data-path="week-4-the-r-environment.html"><a href="week-4-the-r-environment.html#creature-of-the-week-3"><i class="fa fa-check"></i><b>5.13</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html"><i class="fa fa-check"></i><b>6</b> Week 5- R Continued</a>
<ul>
<li class="chapter" data-level="6.1" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#plotting"><i class="fa fa-check"></i><b>6.1</b> 2.1 Plotting</a></li>
<li class="chapter" data-level="6.2" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#scatterplots"><i class="fa fa-check"></i><b>6.2</b> Scatterplots</a></li>
<li class="chapter" data-level="6.3" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#customizing-your-plot"><i class="fa fa-check"></i><b>6.3</b> Customizing your plot</a></li>
<li class="chapter" data-level="6.4" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#practice-problems-2.1"><i class="fa fa-check"></i><b>6.4</b> Practice Problems 2.1</a></li>
<li class="chapter" data-level="6.5" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#plotting-with-ggplot2"><i class="fa fa-check"></i><b>6.5</b> 2.2 plotting with ggplot2</a></li>
<li class="chapter" data-level="6.6" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#practice-problems-2.2"><i class="fa fa-check"></i><b>6.6</b> Practice Problems 2.2</a></li>
<li class="chapter" data-level="6.7" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#for-loops-and-the-apply-family-of-functions"><i class="fa fa-check"></i><b>6.7</b> For loops and the apply family of functions</a></li>
<li class="chapter" data-level="6.8" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#apply-family"><i class="fa fa-check"></i><b>6.8</b> apply family</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#sapply"><i class="fa fa-check"></i><b>6.8.1</b> sapply</a></li>
<li class="chapter" data-level="6.8.2" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#apply"><i class="fa fa-check"></i><b>6.8.2</b> apply</a></li>
<li class="chapter" data-level="6.8.3" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#lapply-list-apply"><i class="fa fa-check"></i><b>6.8.3</b> lapply – “list” apply</a></li>
<li class="chapter" data-level="6.8.4" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#tapply---per-type-apply"><i class="fa fa-check"></i><b>6.8.4</b> tapply - “per Type” apply</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#exercise-2.3-apply-and-tapply"><i class="fa fa-check"></i><b>6.9</b> Exercise 2.3 apply and tapply</a></li>
<li class="chapter" data-level="6.10" data-path="week-5-r-continued.html"><a href="week-5-r-continued.html#creature-of-the-week-4"><i class="fa fa-check"></i><b>6.10</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="week-6-pca.html"><a href="week-6-pca.html"><i class="fa fa-check"></i><b>7</b> Week 6: PCA</a>
<ul>
<li class="chapter" data-level="7.1" data-path="week-6-pca.html"><a href="week-6-pca.html#download-the-data"><i class="fa fa-check"></i><b>7.1</b> Download the data</a></li>
<li class="chapter" data-level="7.2" data-path="week-6-pca.html"><a href="week-6-pca.html#installing-programs"><i class="fa fa-check"></i><b>7.2</b> Installing programs</a></li>
<li class="chapter" data-level="7.3" data-path="week-6-pca.html"><a href="week-6-pca.html#run-pcangsd-on-our-data"><i class="fa fa-check"></i><b>7.3</b> Run pcangsd on our data</a></li>
<li class="chapter" data-level="7.4" data-path="week-6-pca.html"><a href="week-6-pca.html#setting-up-a-new-project-in-r"><i class="fa fa-check"></i><b>7.4</b> Setting up a new project in R</a></li>
<li class="chapter" data-level="7.5" data-path="week-6-pca.html"><a href="week-6-pca.html#reading-data-into-r"><i class="fa fa-check"></i><b>7.5</b> Reading data into R</a></li>
<li class="chapter" data-level="7.6" data-path="week-6-pca.html"><a href="week-6-pca.html#exercises"><i class="fa fa-check"></i><b>7.6</b> Exercises</a></li>
<li class="chapter" data-level="7.7" data-path="week-6-pca.html"><a href="week-6-pca.html#creature-of-the-week-5"><i class="fa fa-check"></i><b>7.7</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html"><i class="fa fa-check"></i><b>8</b> Week 7: F<sub>st</sub> and outlier analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#download-the-data-1"><i class="fa fa-check"></i><b>8.1</b> Download the data</a></li>
<li class="chapter" data-level="8.2" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#getting-r-set-up"><i class="fa fa-check"></i><b>8.2</b> Getting R set up</a></li>
<li class="chapter" data-level="8.3" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#finding-outliers-using-pcadapt"><i class="fa fa-check"></i><b>8.3</b> Finding outliers using pcadapt</a></li>
<li class="chapter" data-level="8.4" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#pcadapt-exercises"><i class="fa fa-check"></i><b>8.4</b> pcadapt Exercises</a></li>
<li class="chapter" data-level="8.5" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#using-fst-to-find-outliers"><i class="fa fa-check"></i><b>8.5</b> Using F<sub>st</sub> to find outliers</a></li>
<li class="chapter" data-level="8.6" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#outflank-practice"><i class="fa fa-check"></i><b>8.6</b> OutFLANK Practice</a></li>
<li class="chapter" data-level="8.7" data-path="week-7-fst-and-outlier-analysis.html"><a href="week-7-fst-and-outlier-analysis.html#creature-of-the-week-6"><i class="fa fa-check"></i><b>8.7</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html"><i class="fa fa-check"></i><b>9</b> Week 8: Genome wide association study (GWAS)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#load-in-the-data"><i class="fa fa-check"></i><b>9.1</b> load in the data</a></li>
<li class="chapter" data-level="9.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-packages-in-r"><i class="fa fa-check"></i><b>9.2</b> install packages in R</a></li>
<li class="chapter" data-level="9.3" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd-and-pcangsd-again"><i class="fa fa-check"></i><b>9.3</b> install angsd and pcangsd (again?)</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd"><i class="fa fa-check"></i><b>9.3.1</b> install angsd</a></li>
<li class="chapter" data-level="9.3.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#for-pcangsd"><i class="fa fa-check"></i><b>9.3.2</b> for pcangsd</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#the-data"><i class="fa fa-check"></i><b>9.4</b> The data</a></li>
<li class="chapter" data-level="9.5" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-vcf-file-and-generate-lrt-file"><i class="fa fa-check"></i><b>9.5</b> take VCF file and generate lrt file</a></li>
<li class="chapter" data-level="9.6" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-lrt-file-and-make-a-manhattan-plot"><i class="fa fa-check"></i><b>9.6</b> take lrt file and make a manhattan plot</a></li>
<li class="chapter" data-level="9.7" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#exercises-1"><i class="fa fa-check"></i><b>9.7</b> Exercises</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Week 6</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="week-6-pca" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Week 6: PCA</h1>
<p>The lecture for this week focused on <strong>Plotting a PCA</strong> can be found <a href="https://github.com/BayLab/MarineGenomics/blob/main/ppt/Week6_lecture.pdf">here</a>.</p>
<p>In week 3 we mapped reads to a genome so we could call our genetic variants (SNPs) and generated bam files for each read file (= sequence data from an individual). Now that we know how to do some R basics we can use those SNPs to plot patterns of population structure via a principal component analysis.</p>
<p>The first portion of this lesson will be performed in <strong>bash</strong> and then we will generate the plots in <strong>RStudio</strong>.</p>
<div id="download-the-data" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Download the data</h2>
<p>We first need to download the data. Use the link below to download it to jetstream, and then us the tar command to un-compress it.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb1-1"><a href="week-6-pca.html#cb1-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2"><a href="week-6-pca.html#cb1-2" aria-hidden="true" tabindex="-1"></a>wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/week6.tar.gz</span>
<span id="cb1-3"><a href="week-6-pca.html#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="week-6-pca.html#cb1-4" aria-hidden="true" tabindex="-1"></a>tar -xzvf week6.tar.gz</span></code></pre></div>
<p>Our data this week consists of genotype_likelihood files in beagle format and a bam.filelist which contains id information for our samples.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb2-1"><a href="week-6-pca.html#cb2-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-2"><a href="week-6-pca.html#cb2-2" aria-hidden="true" tabindex="-1"></a>ls -lh MarineGenomicsData/Week6</span></code></pre></div>
</div>
<div id="installing-programs" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Installing programs</h2>
<p>We will also need to install a few programs. The code below installs the program pcangsd, and a few dependencies for pcangsd:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb3-1"><a href="week-6-pca.html#cb3-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2"><a href="week-6-pca.html#cb3-2" aria-hidden="true" tabindex="-1"></a>#install pip for python</span>
<span id="cb3-3"><a href="week-6-pca.html#cb3-3" aria-hidden="true" tabindex="-1"></a>curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py</span>
<span id="cb3-4"><a href="week-6-pca.html#cb3-4" aria-hidden="true" tabindex="-1"></a>python3 get-pip.py</span>
<span id="cb3-5"><a href="week-6-pca.html#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="week-6-pca.html#cb3-6" aria-hidden="true" tabindex="-1"></a>#we need to add our home directory to the path for pip</span>
<span id="cb3-7"><a href="week-6-pca.html#cb3-7" aria-hidden="true" tabindex="-1"></a>#look at path</span>
<span id="cb3-8"><a href="week-6-pca.html#cb3-8" aria-hidden="true" tabindex="-1"></a>echo $PATH</span>
<span id="cb3-9"><a href="week-6-pca.html#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="week-6-pca.html#cb3-10" aria-hidden="true" tabindex="-1"></a>#add the location of pip to our path</span>
<span id="cb3-11"><a href="week-6-pca.html#cb3-11" aria-hidden="true" tabindex="-1"></a>export PATH=&quot;$HOME/.local/bin:$PATH&quot;</span>
<span id="cb3-12"><a href="week-6-pca.html#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="week-6-pca.html#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="week-6-pca.html#cb3-14" aria-hidden="true" tabindex="-1"></a>#then install pcangsd</span>
<span id="cb3-15"><a href="week-6-pca.html#cb3-15" aria-hidden="true" tabindex="-1"></a>git clone https://github.com/Rosemeis/pcangsd.git</span>
<span id="cb3-16"><a href="week-6-pca.html#cb3-16" aria-hidden="true" tabindex="-1"></a>cd pcangsd/</span>
<span id="cb3-17"><a href="week-6-pca.html#cb3-17" aria-hidden="true" tabindex="-1"></a>pip install --user -r requirements.txt #this installs additional requirements for pcangsd</span>
<span id="cb3-18"><a href="week-6-pca.html#cb3-18" aria-hidden="true" tabindex="-1"></a>python3 setup.py build_ext --inplace</span></code></pre></div>
<p>Check that it’s installed correctly.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb4-1"><a href="week-6-pca.html#cb4-1" aria-hidden="true" tabindex="-1"></a>#navigate to your home directory</span>
<span id="cb4-2"><a href="week-6-pca.html#cb4-2" aria-hidden="true" tabindex="-1"></a>cd</span>
<span id="cb4-3"><a href="week-6-pca.html#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="week-6-pca.html#cb4-4" aria-hidden="true" tabindex="-1"></a>python3 pcangsd/pcangsd.py -h</span></code></pre></div>
<p>You should see the help menu for pcangsd.</p>
</div>
<div id="run-pcangsd-on-our-data" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Run pcangsd on our data</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb5-1"><a href="week-6-pca.html#cb5-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-2"><a href="week-6-pca.html#cb5-2" aria-hidden="true" tabindex="-1"></a>python3 ../../pcangsd/pcangsd.py -beagle genolike_beagle.beagle.gz -o pca_out -threads 28</span></code></pre></div>
<p>This will generate a covariance matrix output file. We can read this file into R and compute the eigenvectors and eigenvalues and make our plot.</p>
</div>
<div id="setting-up-a-new-project-in-r" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> Setting up a new project in R</h2>
<p>We’ll first open a new <code>.R</code> script to keep track of our code and everything that we’re doing. Be sure to comment heavily with <code>#</code>. It may seem obvious what we’re doing now, but you will forget what certain lines of code do in a surprisingly short amount of time.</p>
</div>
<div id="reading-data-into-r" class="section level2" number="7.5">
<h2><span class="header-section-number">7.5</span> Reading data into R</h2>
<p>There are several ways to read data in R. Useful functions include <code>read.table()</code>, <code>getwd()</code>, and <code>setwd()</code></p>
<p>R like many other programs is set to start in a specific direcotory, usually the desktop or homw directory. You can check and see what directory R is set in with the command <code>getwd()</code></p>
<p>And of course all of this code is typed into our script and commented along the way.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="week-6-pca.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#check working directory</span></span>
<span id="cb6-2"><a href="week-6-pca.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#getwd()</span></span>
<span id="cb6-3"><a href="week-6-pca.html#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="week-6-pca.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#should output</span></span>
<span id="cb6-5"><a href="week-6-pca.html#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="week-6-pca.html#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;/home/margeno&quot;</span></span></code></pre></div>
<p>Thus in jetstream the working directory is set to our home directory, which will work fine for uploading data. If you need to change the directory you can use <code>setwd()</code> and provide it with the full directory path (e.g., "C:") and then confirm that it moved the directory with <code>getwd()</code>.</p>
<p>In Rstudio you can also use the pull down menus <code>Session</code> to get and set your working directory. Similarly, the tab <code>Files</code> in the lower right will also allow you to set your working directory.</p>
<p>Now well read the data output by angsd into R using <code>read.table()</code> and <code>as.matrix()</code></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="week-6-pca.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#cov&lt;-as.matrix(read.table(&quot;MarineGenomicsData/Week6/pca_out.cov&quot;))</span></span></code></pre></div>
<p>And then we can compute the eigenvalues from our covariance matrix with the function <code>eigen</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="week-6-pca.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#e&lt;-eigen(cov)</span></span></code></pre></div>
<p>And make a simple plot in base R</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="week-6-pca.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#plot(e$vectors[,1:2])</span></span></code></pre></div>
<p>We may be interested in how much of the variance our first two components explain. You can look at this in the <code>e</code> object under values. We divide the eigen values by the sum of all the values to get the percent explained by each value.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="week-6-pca.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#e$values/sum(e$values)</span></span></code></pre></div>
<p>We now want to make the colors match population labels. The information for which population each individual sample came from is in the bam.filelist file.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="week-6-pca.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data</span></span>
<span id="cb11-2"><a href="week-6-pca.html#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="week-6-pca.html#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#names&lt;-read.table(&quot;MarineGenomicsData/Week6/bam.filelist&quot;)</span></span>
<span id="cb11-4"><a href="week-6-pca.html#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="week-6-pca.html#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#assign the rownames of the covariance matix the </span></span>
<span id="cb11-6"><a href="week-6-pca.html#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="week-6-pca.html#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co">#rownames(cov)&lt;-names$V2</span></span>
<span id="cb11-8"><a href="week-6-pca.html#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="week-6-pca.html#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co">#remake the plot with the colors we want</span></span>
<span id="cb11-10"><a href="week-6-pca.html#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="week-6-pca.html#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co">#plot(e$vectors[,1:2], col=as.factor(rownames(cov)), pch=16)</span></span></code></pre></div>
<p>There is another column in the bam.filelist file that will allow us to color the populations based on region. Go ahead and modify your plots to have nice x and y labels that state the percent variance explained by each axis, and that are colored by region. You can also change the pch and any other parameters you would like.</p>
</div>
<div id="exercises" class="section level2" number="7.6">
<h2><span class="header-section-number">7.6</span> Exercises</h2>
<blockquote>
<h1 id="practice-questions">Practice Questions</h1>
</blockquote>
<blockquote>
<ol style="list-style-type: decimal">
<li>In R, upload the genotype likelihood maf file (genolike.beagle.maf.gz). Filter the data so that we only use sites for which all 15 individuals have data present (i.e., subsite by the nInd column). How many SNPs does this leave us? How many SNPs did we start with?
Hint use the command <code>read.table(gzfile("path/to/genotypelikelihoodfile/), header=T)</code> to read in the file in R. Dont forget to assign it to an object or R will print the whole thing to the screen.</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: goldenrod;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb12"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb12-1"><a href="week-6-pca.html#cb12-1" aria-hidden="true" tabindex="-1"></a># read in the data</span>
<span id="cb12-2"><a href="week-6-pca.html#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="week-6-pca.html#cb12-3" aria-hidden="true" tabindex="-1"></a>#### read in genotype likelihood data and filter by all individuals</span>
<span id="cb12-4"><a href="week-6-pca.html#cb12-4" aria-hidden="true" tabindex="-1"></a>gen<span class="er">&lt;</span>-read.table(gzfile(&quot;MarineGenomicsData/Week6/genolike_beagle.mafs.gz&quot;), header=T)</span>
<span id="cb12-5"><a href="week-6-pca.html#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="week-6-pca.html#cb12-6" aria-hidden="true" tabindex="-1"></a>#filter by nind == 15</span>
<span id="cb12-7"><a href="week-6-pca.html#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="week-6-pca.html#cb12-8" aria-hidden="true" tabindex="-1"></a>gen_allIND<span class="er">&lt;</span>-gen[gen$nInd==15,]</span>
<span id="cb12-9"><a href="week-6-pca.html#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="week-6-pca.html#cb12-10" aria-hidden="true" tabindex="-1"></a>#find how many SNPs we had in the first dataset</span>
<span id="cb12-11"><a href="week-6-pca.html#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="week-6-pca.html#cb12-12" aria-hidden="true" tabindex="-1"></a>dim(gen)</span>
<span id="cb12-13"><a href="week-6-pca.html#cb12-13" aria-hidden="true" tabindex="-1"></a>#[1] 282950      7</span>
<span id="cb12-14"><a href="week-6-pca.html#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="week-6-pca.html#cb12-15" aria-hidden="true" tabindex="-1"></a>#how many do we have in the filtered dataset?</span>
<span id="cb12-16"><a href="week-6-pca.html#cb12-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-17"><a href="week-6-pca.html#cb12-17" aria-hidden="true" tabindex="-1"></a>dim(gen_allIND)</span>
<span id="cb12-18"><a href="week-6-pca.html#cb12-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-19"><a href="week-6-pca.html#cb12-19" aria-hidden="true" tabindex="-1"></a>#[1] 31905     7</span></code></pre></div>
</p>
</details>
<blockquote>
<ol start="2" style="list-style-type: decimal">
<li>Use the filtered file from 01 in pcangsd to recalculate the covarianve matrix and regenerate the pca plot. How does it differ from the one we generated in class?
Hint this will require you to merge the filtered data frame from exercise 1 with the beagle formated dataframe. Use the function <code>merge(x, y, by = )</code>, second hint, the function merge only works (properly) if the column names are the same. Change the column names to match across the two dataframes where they have the same info.</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: goldenrod;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb13"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb13-1"><a href="week-6-pca.html#cb13-1" aria-hidden="true" tabindex="-1"></a>## read in the beagle formatted data</span>
<span id="cb13-2"><a href="week-6-pca.html#cb13-2" aria-hidden="true" tabindex="-1"></a>beag<span class="er">&lt;</span>-read.table(gzfile(&quot;MarineGenomicsData/Week6/genolike_beagle.beagle.gz&quot;), header=T)</span>
<span id="cb13-3"><a href="week-6-pca.html#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="week-6-pca.html#cb13-4" aria-hidden="true" tabindex="-1"></a>#look at the column names between the two datasets</span>
<span id="cb13-5"><a href="week-6-pca.html#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="week-6-pca.html#cb13-6" aria-hidden="true" tabindex="-1"></a>colnames(gen)</span>
<span id="cb13-7"><a href="week-6-pca.html#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="week-6-pca.html#cb13-8" aria-hidden="true" tabindex="-1"></a>colnames(beag)</span>
<span id="cb13-9"><a href="week-6-pca.html#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="week-6-pca.html#cb13-10" aria-hidden="true" tabindex="-1"></a>#annoyingly the beagle formated data merges the chromo and position columns of the maf dataframe</span>
<span id="cb13-11"><a href="week-6-pca.html#cb13-11" aria-hidden="true" tabindex="-1"></a>#two solutions: 1. we could split the columns in beag to match chromo and position</span>
<span id="cb13-12"><a href="week-6-pca.html#cb13-12" aria-hidden="true" tabindex="-1"></a>#               2. we merge the columns in gen to match the format in beag</span>
<span id="cb13-13"><a href="week-6-pca.html#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="week-6-pca.html#cb13-14" aria-hidden="true" tabindex="-1"></a>#since we need the beagle format to run pcangsd lets do option 2. </span>
<span id="cb13-15"><a href="week-6-pca.html#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="week-6-pca.html#cb13-16" aria-hidden="true" tabindex="-1"></a>#we&#39;ll use the paste() function to to this</span>
<span id="cb13-17"><a href="week-6-pca.html#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="week-6-pca.html#cb13-18" aria-hidden="true" tabindex="-1"></a>#we&#39;ll make a new column in gen that is labeled marker to match the beagle column</span>
<span id="cb13-19"><a href="week-6-pca.html#cb13-19" aria-hidden="true" tabindex="-1"></a>#   gen_allIND$marker<span class="er">&lt;</span>-paste(gen_allIND$chromo, gen_allIND$position, sep=&quot;_&quot;)</span>
<span id="cb13-20"><a href="week-6-pca.html#cb13-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-21"><a href="week-6-pca.html#cb13-21" aria-hidden="true" tabindex="-1"></a>#then merge them by the marker column</span>
<span id="cb13-22"><a href="week-6-pca.html#cb13-22" aria-hidden="true" tabindex="-1"></a>beag_allIND<span class="er">&lt;</span>-merge(gen_allIND, beag, by=&quot;marker&quot;)</span>
<span id="cb13-23"><a href="week-6-pca.html#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="week-6-pca.html#cb13-24" aria-hidden="true" tabindex="-1"></a>#that should return a dataframe with 31905 rows and 55 columns</span>
<span id="cb13-25"><a href="week-6-pca.html#cb13-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-26"><a href="week-6-pca.html#cb13-26" aria-hidden="true" tabindex="-1"></a>#pcangsd doesn&#39;t want those extra columns that came from our merge</span>
<span id="cb13-27"><a href="week-6-pca.html#cb13-27" aria-hidden="true" tabindex="-1"></a>#lets get rid of them to leave the 48 columns that beagle needs.</span>
<span id="cb13-28"><a href="week-6-pca.html#cb13-28" aria-hidden="true" tabindex="-1"></a>#use colnames to see which ones we want</span>
<span id="cb13-29"><a href="week-6-pca.html#cb13-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-30"><a href="week-6-pca.html#cb13-30" aria-hidden="true" tabindex="-1"></a>colnames(beag_allIND)</span>
<span id="cb13-31"><a href="week-6-pca.html#cb13-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-32"><a href="week-6-pca.html#cb13-32" aria-hidden="true" tabindex="-1"></a>#we want the first column and then columns 9-55</span>
<span id="cb13-33"><a href="week-6-pca.html#cb13-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-34"><a href="week-6-pca.html#cb13-34" aria-hidden="true" tabindex="-1"></a>beag_allIND_final<span class="er">&lt;</span>-beag_allIND[,c(1,9:55)]</span>
<span id="cb13-35"><a href="week-6-pca.html#cb13-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-36"><a href="week-6-pca.html#cb13-36" aria-hidden="true" tabindex="-1"></a>#now we have a dataframe with 31905 rows and 48 columns</span>
<span id="cb13-37"><a href="week-6-pca.html#cb13-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-38"><a href="week-6-pca.html#cb13-38" aria-hidden="true" tabindex="-1"></a>#let&#39;s write a table that we can use to run in pcangsd</span>
<span id="cb13-39"><a href="week-6-pca.html#cb13-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-40"><a href="week-6-pca.html#cb13-40" aria-hidden="true" tabindex="-1"></a>write.table(beag_allIND_final, &quot;MarineGenomicsData/Week6/geno_like_filt_allIND.beagle&quot;, </span>
<span id="cb13-41"><a href="week-6-pca.html#cb13-41" aria-hidden="true" tabindex="-1"></a>            sep=&quot;\t&quot;,</span>
<span id="cb13-42"><a href="week-6-pca.html#cb13-42" aria-hidden="true" tabindex="-1"></a>            row.names = F)#bydefault write.table outputs a dataframe with row names we dont want that</span>
<span id="cb13-43"><a href="week-6-pca.html#cb13-43" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-44"><a href="week-6-pca.html#cb13-44" aria-hidden="true" tabindex="-1"></a>### This writes a file to our Marine Genomics data directory. </span>
<span id="cb13-45"><a href="week-6-pca.html#cb13-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-46"><a href="week-6-pca.html#cb13-46" aria-hidden="true" tabindex="-1"></a># pcangsd needs the file to be gzipped</span>
<span id="cb13-47"><a href="week-6-pca.html#cb13-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-48"><a href="week-6-pca.html#cb13-48" aria-hidden="true" tabindex="-1"></a>$ gzip geno_like_filt_allIND.beagle</span>
<span id="cb13-49"><a href="week-6-pca.html#cb13-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-50"><a href="week-6-pca.html#cb13-50" aria-hidden="true" tabindex="-1"></a># and now we can run our pcangsd code</span>
<span id="cb13-51"><a href="week-6-pca.html#cb13-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-52"><a href="week-6-pca.html#cb13-52" aria-hidden="true" tabindex="-1"></a>$ python3 ../../pcangsd/pcangsd.py -beagle geno_like_filt_allIND.beagle.gz -o pca_out_allind -threads 28</span>
<span id="cb13-53"><a href="week-6-pca.html#cb13-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-54"><a href="week-6-pca.html#cb13-54" aria-hidden="true" tabindex="-1"></a># and then we can go back to R and rerun our code to calulate the eigenvalues and eigenvectors for this covariance matrix. </span>
<span id="cb13-55"><a href="week-6-pca.html#cb13-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-56"><a href="week-6-pca.html#cb13-56" aria-hidden="true" tabindex="-1"></a>#read in the new covariance matrix                               </span>
<span id="cb13-57"><a href="week-6-pca.html#cb13-57" aria-hidden="true" tabindex="-1"></a>cov_allind<span class="er">&lt;</span>-as.matrix(read.table(&quot;MarineGenomicsData/Week6/pca_out_allind.cov&quot;))</span>
<span id="cb13-58"><a href="week-6-pca.html#cb13-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-59"><a href="week-6-pca.html#cb13-59" aria-hidden="true" tabindex="-1"></a>#calculate eigenvalues</span>
<span id="cb13-60"><a href="week-6-pca.html#cb13-60" aria-hidden="true" tabindex="-1"></a>e_allind<span class="er">&lt;</span>-eigen(cov_allind)</span>
<span id="cb13-61"><a href="week-6-pca.html#cb13-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-62"><a href="week-6-pca.html#cb13-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-63"><a href="week-6-pca.html#cb13-63" aria-hidden="true" tabindex="-1"></a>#plot the data</span>
<span id="cb13-64"><a href="week-6-pca.html#cb13-64" aria-hidden="true" tabindex="-1"></a>plot(e_allind$vectors[,1:2])</span>
<span id="cb13-65"><a href="week-6-pca.html#cb13-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-66"><a href="week-6-pca.html#cb13-66" aria-hidden="true" tabindex="-1"></a>#how much variance does our pc explain</span>
<span id="cb13-67"><a href="week-6-pca.html#cb13-67" aria-hidden="true" tabindex="-1"></a>e_allind$values/sum(e_allind$values)                              </span>
<span id="cb13-68"><a href="week-6-pca.html#cb13-68" aria-hidden="true" tabindex="-1"></a>                               </span>
<span id="cb13-69"><a href="week-6-pca.html#cb13-69" aria-hidden="true" tabindex="-1"></a>                               </span></code></pre></div>
</p>
</details>
<blockquote>
<ol start="3" style="list-style-type: decimal">
<li>Remake the base R plot that we generated in class in ggplot. Use the group parameter in aes to group by region and/or population. Note, ggplot wants the data to be a dataframe and our data is a matrix. Use the function <code>as.data.frame(e$vectors)</code> to acheive this in the ggplot function. Don’t forget to load the ggplot2 package with `library(ggplot2)</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: goldenrod;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb14"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb14-1"><a href="week-6-pca.html#cb14-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-2"><a href="week-6-pca.html#cb14-2" aria-hidden="true" tabindex="-1"></a># we need to use the function as.data.frame to make our matrix a dataframe.</span>
<span id="cb14-3"><a href="week-6-pca.html#cb14-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-4"><a href="week-6-pca.html#cb14-4" aria-hidden="true" tabindex="-1"></a>ggplot(as.data.frame(e$vectors), aes(V1, V2, group=as.factor(names$V3), color=as.factor(names$V3)))+</span>
<span id="cb14-5"><a href="week-6-pca.html#cb14-5" aria-hidden="true" tabindex="-1"></a>  geom_point()+</span>
<span id="cb14-6"><a href="week-6-pca.html#cb14-6" aria-hidden="true" tabindex="-1"></a>  theme_classic()+</span>
<span id="cb14-7"><a href="week-6-pca.html#cb14-7" aria-hidden="true" tabindex="-1"></a>  scale_color_manual(name=&quot;Region&quot;, </span>
<span id="cb14-8"><a href="week-6-pca.html#cb14-8" aria-hidden="true" tabindex="-1"></a>                     labels = c(&quot;North&quot;, </span>
<span id="cb14-9"><a href="week-6-pca.html#cb14-9" aria-hidden="true" tabindex="-1"></a>                                &quot;South&quot;), </span>
<span id="cb14-10"><a href="week-6-pca.html#cb14-10" aria-hidden="true" tabindex="-1"></a>                     values = c(&quot;a&quot;=&quot;slateblue&quot;, </span>
<span id="cb14-11"><a href="week-6-pca.html#cb14-11" aria-hidden="true" tabindex="-1"></a>                                &quot;b&quot;=&quot;green&quot;))  </span>
<span id="cb14-12"><a href="week-6-pca.html#cb14-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-13"><a href="week-6-pca.html#cb14-13" aria-hidden="true" tabindex="-1"></a></span></code></pre></div>
</p>
</details>
</div>
<div id="creature-of-the-week-5" class="section level2" number="7.7">
<h2><span class="header-section-number">7.7</span> Creature of the Week!</h2>
<div class="figure">
<img src="figs/creatures/boringsponge.jpg" alt="" />
<p class="caption">Yellow Boring Sponge, the inspiration for Spongebob! (Image credit: Bernard Picton/Encyclopedia of Marine Life of Britain and Ireland)</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="week-5-r-continued.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="week-7-fst-and-outlier-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Test.pdf", "Test.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
