<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>12 Week 10- Genomics is a BLAST | Test.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="12 Week 10- Genomics is a BLAST | Test.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="12 Week 10- Genomics is a BLAST | Test.knit" />
  
  
  

<meta name="author" content="" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="week-9--population-structure-using-ngsadmix.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Marine Genomics 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to Marine Genomics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#cloud-computing-versus-individual-laptops"><i class="fa fa-check"></i><b>1.1</b> Cloud computing versus individual laptops</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#meet-our-instructors"><i class="fa fa-check"></i><b>1.2</b> Meet our instructors!</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.3</b> Schedule</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="week-1--welcome.html"><a href="week-1--welcome.html"><i class="fa fa-check"></i><b>2</b> Week 1- Welcome!</a>
<ul>
<li class="chapter" data-level="2.1" data-path="week-1--welcome.html"><a href="week-1--welcome.html#introduction-to-shell-computing-via-the-data-carpentry-tutorial"><i class="fa fa-check"></i><b>2.1</b> Introduction to shell computing via the data carpentry tutorial</a></li>
<li class="chapter" data-level="2.2" data-path="week-1--welcome.html"><a href="week-1--welcome.html#how-to-access-the-shell"><i class="fa fa-check"></i><b>2.2</b> How to access the shell</a></li>
<li class="chapter" data-level="2.3" data-path="week-1--welcome.html"><a href="week-1--welcome.html#week-1-objectives"><i class="fa fa-check"></i><b>2.3</b> Week 1 Objectives</a></li>
<li class="chapter" data-level="2.4" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-your-file-system"><i class="fa fa-check"></i><b>2.4</b> Navigating your file system</a></li>
<li class="chapter" data-level="2.5" data-path="week-1--welcome.html"><a href="week-1--welcome.html#shortcut-tab-completion"><i class="fa fa-check"></i><b>2.5</b> Shortcut: Tab Completion</a></li>
<li class="chapter" data-level="2.6" data-path="week-1--welcome.html"><a href="week-1--welcome.html#summary-key-points"><i class="fa fa-check"></i><b>2.6</b> Summary &amp; Key Points</a></li>
<li class="chapter" data-level="2.7" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-files-and-directories"><i class="fa fa-check"></i><b>2.7</b> Navigating Files and Directories</a></li>
<li class="chapter" data-level="2.8" data-path="week-1--welcome.html"><a href="week-1--welcome.html#moving-around-the-file-system"><i class="fa fa-check"></i><b>2.8</b> Moving around the file system</a></li>
<li class="chapter" data-level="2.9" data-path="week-1--welcome.html"><a href="week-1--welcome.html#examining-the-contents-of-other-directories"><i class="fa fa-check"></i><b>2.9</b> Examining the contents of other directories</a></li>
<li class="chapter" data-level="2.10" data-path="week-1--welcome.html"><a href="week-1--welcome.html#full-vs-relative-paths"><i class="fa fa-check"></i><b>2.10</b> Full vs Relative Paths</a></li>
<li class="chapter" data-level="2.11" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigational-shortcuts"><i class="fa fa-check"></i><b>2.11</b> Navigational shortcuts</a></li>
<li class="chapter" data-level="2.12" data-path="week-1--welcome.html"><a href="week-1--welcome.html#key-points"><i class="fa fa-check"></i><b>2.12</b> Key Points</a></li>
<li class="chapter" data-level="2.13" data-path="week-1--welcome.html"><a href="week-1--welcome.html#creature-of-the-week"><i class="fa fa-check"></i><b>2.13</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html"><i class="fa fa-check"></i><b>3</b> Week 2-Working With Files</a></li>
<li class="chapter" data-level="4" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><i class="fa fa-check"></i><b>4</b> Before we start, make sure to clone or update the github folder MarineGenomics in the user directory</a>
<ul>
<li class="chapter" data-level="4.1" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#our-data-set-fastq-files"><i class="fa fa-check"></i><b>4.1</b> Our data set: FASTQ files</a></li>
<li class="chapter" data-level="4.2" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#wildcards"><i class="fa fa-check"></i><b>4.2</b> Wildcards</a></li>
<li class="chapter" data-level="4.3" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#command-history"><i class="fa fa-check"></i><b>4.3</b> Command History</a></li>
<li class="chapter" data-level="4.4" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#examining-files"><i class="fa fa-check"></i><b>4.4</b> Examining Files</a></li>
<li class="chapter" data-level="4.5" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#details-on-the-fastq-format"><i class="fa fa-check"></i><b>4.5</b> Details on the FASTQ format</a></li>
<li class="chapter" data-level="4.6" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#creating-moving-copying-and-removing"><i class="fa fa-check"></i><b>4.6</b> Creating, moving, copying, and removing</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#copying-files"><i class="fa fa-check"></i><b>4.6.1</b> Copying Files</a></li>
<li class="chapter" data-level="4.6.2" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#creating-directories"><i class="fa fa-check"></i><b>4.6.2</b> Creating Directories</a></li>
<li class="chapter" data-level="4.6.3" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#moving-renaming"><i class="fa fa-check"></i><b>4.6.3</b> Moving / Renaming</a></li>
<li class="chapter" data-level="4.6.4" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#file-permissions"><i class="fa fa-check"></i><b>4.6.4</b> File Permissions</a></li>
<li class="chapter" data-level="4.6.5" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#removing"><i class="fa fa-check"></i><b>4.6.5</b> Removing</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#redirections"><i class="fa fa-check"></i><b>4.7</b> Redirections</a></li>
<li class="chapter" data-level="4.8" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#searching-files"><i class="fa fa-check"></i><b>4.8</b> Searching files</a></li>
<li class="chapter" data-level="4.9" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#redirecting-output"><i class="fa fa-check"></i><b>4.9</b> Redirecting output</a></li>
<li class="chapter" data-level="4.10" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#writing-for-loops"><i class="fa fa-check"></i><b>4.10</b> Writing for loops</a></li>
<li class="chapter" data-level="4.11" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#using-basename-in-for-loops"><i class="fa fa-check"></i><b>4.11</b> Using Basename in for loops</a></li>
<li class="chapter" data-level="4.12" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#writing-scripts-and-working-with-data"><i class="fa fa-check"></i><b>4.12</b> Writing Scripts and Working with Data</a></li>
<li class="chapter" data-level="4.13" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#writing-files"><i class="fa fa-check"></i><b>4.13</b> Writing files</a></li>
<li class="chapter" data-level="4.14" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#writing-scripts"><i class="fa fa-check"></i><b>4.14</b> Writing scripts</a></li>
<li class="chapter" data-level="4.15" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#moving-and-downloading-data"><i class="fa fa-check"></i><b>4.15</b> Moving and Downloading Data</a>
<ul>
<li class="chapter" data-level="4.15.1" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#getting-data-from-the-cloud"><i class="fa fa-check"></i><b>4.15.1</b> Getting data from the cloud</a></li>
<li class="chapter" data-level="4.15.2" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix"><i class="fa fa-check"></i><b>4.15.2</b> Uploading and Downloading Data to your Virtual Machine with scp - UNIX</a></li>
</ul></li>
<li class="chapter" data-level="4.16" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#awk"><i class="fa fa-check"></i><b>4.16</b> Awk</a></li>
<li class="chapter" data-level="4.17" data-path="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html"><a href="before-we-start-make-sure-to-clone-or-update-the-github-folder-marinegenomics-in-the-user-directory.html#creature-of-the-week-1"><i class="fa fa-check"></i><b>4.17</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html"><i class="fa fa-check"></i><b>5</b> Week 3- What is a Genetic Variant?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#to-get-started-lets-dowload-the-data-and-install-a-few-programs"><i class="fa fa-check"></i><b>5.1</b> To get started lets dowload the data and install a few programs</a></li>
<li class="chapter" data-level="5.2" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#raw-read-quality-control"><i class="fa fa-check"></i><b>5.2</b> Raw read quality control</a></li>
<li class="chapter" data-level="5.3" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#trimming-to-remove-adapters"><i class="fa fa-check"></i><b>5.3</b> Trimming to remove adapters</a></li>
<li class="chapter" data-level="5.4" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#building-an-index-of-our-genome"><i class="fa fa-check"></i><b>5.4</b> Building an index of our genome</a></li>
<li class="chapter" data-level="5.5" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#map-reads-to-the-genome"><i class="fa fa-check"></i><b>5.5</b> Map reads to the genome</a></li>
<li class="chapter" data-level="5.6" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#sam-to-bam-file-conversion"><i class="fa fa-check"></i><b>5.6</b> sam to bam file conversion</a></li>
<li class="chapter" data-level="5.7" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#genotype-likelihoods"><i class="fa fa-check"></i><b>5.7</b> Genotype likelihoods</a></li>
<li class="chapter" data-level="5.8" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#creature-of-the-week-2"><i class="fa fa-check"></i><b>5.8</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html"><i class="fa fa-check"></i><b>6</b> Week 4- The R environment</a>
<ul>
<li class="chapter" data-level="6.1" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#lesson-1-orientation-to-r"><i class="fa fa-check"></i><b>6.1</b> Lesson 1: Orientation to R</a></li>
<li class="chapter" data-level="6.2" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#manipulating-a-vector-object"><i class="fa fa-check"></i><b>6.2</b> Manipulating a vector object</a></li>
<li class="chapter" data-level="6.3" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#operations-act-on-each-element-of-a-vector"><i class="fa fa-check"></i><b>6.3</b> Operations act on each element of a vector:</a></li>
<li class="chapter" data-level="6.4" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#operations-can-also-work-with-two-vectors"><i class="fa fa-check"></i><b>6.4</b> Operations can also work with two vectors:</a></li>
<li class="chapter" data-level="6.5" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#a-few-tips-below-for-working-with-objects"><i class="fa fa-check"></i><b>6.5</b> A few tips below for working with objects:</a></li>
<li class="chapter" data-level="6.6" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#practice-r-operations"><i class="fa fa-check"></i><b>6.6</b> Practice R Operations</a></li>
<li class="chapter" data-level="6.7" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#characterizing-a-dataframe"><i class="fa fa-check"></i><b>6.7</b> 1.2 Characterizing a dataframe</a></li>
<li class="chapter" data-level="6.8" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#how-to-access-parts-of-the-data"><i class="fa fa-check"></i><b>6.8</b> How to access parts of the data:</a></li>
<li class="chapter" data-level="6.9" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#data-manipulation"><i class="fa fa-check"></i><b>6.9</b> Data Manipulation</a></li>
<li class="chapter" data-level="6.10" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#practice-exploring-a-dataframe"><i class="fa fa-check"></i><b>6.10</b> Practice exploring a dataframe</a></li>
<li class="chapter" data-level="6.11" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#subsetting-datasets-logicals"><i class="fa fa-check"></i><b>6.11</b> 1.3 Subsetting datasets &amp; logicals</a></li>
<li class="chapter" data-level="6.12" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#practice-subsetting-datasetslogicals"><i class="fa fa-check"></i><b>6.12</b> Practice Subsetting datasets/logicals</a></li>
<li class="chapter" data-level="6.13" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#creature-of-the-week-3"><i class="fa fa-check"></i><b>6.13</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html"><i class="fa fa-check"></i><b>7</b> Week 5- R Continued</a>
<ul>
<li class="chapter" data-level="7.1" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#plotting"><i class="fa fa-check"></i><b>7.1</b> 2.1 Plotting</a></li>
<li class="chapter" data-level="7.2" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#scatterplots"><i class="fa fa-check"></i><b>7.2</b> Scatterplots</a></li>
<li class="chapter" data-level="7.3" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#customizing-your-plot"><i class="fa fa-check"></i><b>7.3</b> Customizing your plot</a></li>
<li class="chapter" data-level="7.4" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#practice-problems-2.1"><i class="fa fa-check"></i><b>7.4</b> Practice Problems 2.1</a></li>
<li class="chapter" data-level="7.5" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#plotting-with-ggplot2"><i class="fa fa-check"></i><b>7.5</b> 2.2 plotting with ggplot2</a></li>
<li class="chapter" data-level="7.6" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#practice-problems-2.2"><i class="fa fa-check"></i><b>7.6</b> Practice Problems 2.2</a></li>
<li class="chapter" data-level="7.7" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#for-loops-and-the-apply-family-of-functions"><i class="fa fa-check"></i><b>7.7</b> For loops and the apply family of functions</a></li>
<li class="chapter" data-level="7.8" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#apply-family"><i class="fa fa-check"></i><b>7.8</b> apply family</a>
<ul>
<li class="chapter" data-level="7.8.1" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#sapply"><i class="fa fa-check"></i><b>7.8.1</b> sapply</a></li>
<li class="chapter" data-level="7.8.2" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#apply"><i class="fa fa-check"></i><b>7.8.2</b> apply</a></li>
<li class="chapter" data-level="7.8.3" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#lapply-list-apply"><i class="fa fa-check"></i><b>7.8.3</b> lapply – “list” apply</a></li>
<li class="chapter" data-level="7.8.4" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#tapply---per-type-apply"><i class="fa fa-check"></i><b>7.8.4</b> tapply - “per Type” apply</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#exercise-2.3-apply-and-tapply"><i class="fa fa-check"></i><b>7.9</b> Exercise 2.3 apply and tapply</a></li>
<li class="chapter" data-level="7.10" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#creature-of-the-week-4"><i class="fa fa-check"></i><b>7.10</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html"><i class="fa fa-check"></i><b>8</b> Week 6- Principle Component Analyses</a>
<ul>
<li class="chapter" data-level="8.1" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#download-the-data"><i class="fa fa-check"></i><b>8.1</b> Download the data</a></li>
<li class="chapter" data-level="8.2" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#installing-programs"><i class="fa fa-check"></i><b>8.2</b> Installing programs</a></li>
<li class="chapter" data-level="8.3" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#run-pcangsd-on-our-data"><i class="fa fa-check"></i><b>8.3</b> Run pcangsd on our data</a></li>
<li class="chapter" data-level="8.4" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#setting-up-a-new-project-in-r"><i class="fa fa-check"></i><b>8.4</b> Setting up a new project in R</a></li>
<li class="chapter" data-level="8.5" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#reading-data-into-r"><i class="fa fa-check"></i><b>8.5</b> Reading data into R</a></li>
<li class="chapter" data-level="8.6" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#exercises"><i class="fa fa-check"></i><b>8.6</b> Exercises</a></li>
<li class="chapter" data-level="8.7" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#creature-of-the-week-5"><i class="fa fa-check"></i><b>8.7</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html"><i class="fa fa-check"></i><b>9</b> Week 7- F<sub>st</sub> and outlier analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#download-the-data-1"><i class="fa fa-check"></i><b>9.1</b> Download the data</a></li>
<li class="chapter" data-level="9.2" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#getting-r-set-up"><i class="fa fa-check"></i><b>9.2</b> Getting R set up</a></li>
<li class="chapter" data-level="9.3" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#finding-outliers-using-pcadapt"><i class="fa fa-check"></i><b>9.3</b> Finding outliers using pcadapt</a></li>
<li class="chapter" data-level="9.4" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#pcadapt-exercises"><i class="fa fa-check"></i><b>9.4</b> pcadapt Exercises</a></li>
<li class="chapter" data-level="9.5" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#using-fst-to-find-outliers"><i class="fa fa-check"></i><b>9.5</b> Using F<sub>st</sub> to find outliers</a></li>
<li class="chapter" data-level="9.6" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#outflank-practice"><i class="fa fa-check"></i><b>9.6</b> OutFLANK Practice</a></li>
<li class="chapter" data-level="9.7" data-path="week-7--fst-and-outlier-analysis.html"><a href="week-7--fst-and-outlier-analysis.html#creature-of-the-week-6"><i class="fa fa-check"></i><b>9.7</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html"><i class="fa fa-check"></i><b>10</b> Week 8: Genome wide association study (GWAS)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#load-in-the-data"><i class="fa fa-check"></i><b>10.1</b> load in the data</a></li>
<li class="chapter" data-level="10.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-packages-in-r"><i class="fa fa-check"></i><b>10.2</b> install packages in R</a></li>
<li class="chapter" data-level="10.3" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd-and-pcangsd-again"><i class="fa fa-check"></i><b>10.3</b> install angsd and pcangsd (again?)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd"><i class="fa fa-check"></i><b>10.3.1</b> install angsd</a></li>
<li class="chapter" data-level="10.3.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#for-pcangsd"><i class="fa fa-check"></i><b>10.3.2</b> for pcangsd</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#the-data"><i class="fa fa-check"></i><b>10.4</b> The data</a></li>
<li class="chapter" data-level="10.5" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-beagle-file-and-generate-lrt-file"><i class="fa fa-check"></i><b>10.5</b> take Beagle file and generate lrt file</a></li>
<li class="chapter" data-level="10.6" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-lrt-file-and-make-a-manhattan-plot"><i class="fa fa-check"></i><b>10.6</b> take lrt file and make a manhattan plot</a></li>
<li class="chapter" data-level="10.7" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#exercises-1"><i class="fa fa-check"></i><b>10.7</b> Exercises</a></li>
<li class="chapter" data-level="10.8" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#creature-of-the-week-7"><i class="fa fa-check"></i><b>10.8</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html"><i class="fa fa-check"></i><b>11</b> Week 9- Population Structure using NGSadmix</a>
<ul>
<li class="chapter" data-level="11.1" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#download-the-data-2"><i class="fa fa-check"></i><b>11.1</b> Download the data</a></li>
<li class="chapter" data-level="11.2" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#using-ngsadmix"><i class="fa fa-check"></i><b>11.2</b> Using NGSadmix</a></li>
<li class="chapter" data-level="11.3" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#ploting-stucture-in-r"><i class="fa fa-check"></i><b>11.3</b> ploting stucture in R</a></li>
<li class="chapter" data-level="11.4" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#how-do-we-know-which-k-to-pick"><i class="fa fa-check"></i><b>11.4</b> How do we know which K to pick?</a></li>
<li class="chapter" data-level="11.5" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#exercises-2"><i class="fa fa-check"></i><b>11.5</b> Exercises</a></li>
<li class="chapter" data-level="11.6" data-path="week-9--population-structure-using-ngsadmix.html"><a href="week-9--population-structure-using-ngsadmix.html#creature-of-the-week-8"><i class="fa fa-check"></i><b>11.6</b> Creature of the Week!</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html"><i class="fa fa-check"></i><b>12</b> Week 10- Genomics is a BLAST</a>
<ul>
<li class="chapter" data-level="12.1" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#genome-annotation-and-blast-searches"><i class="fa fa-check"></i><b>12.1</b> Genome Annotation and BLAST searches</a></li>
<li class="chapter" data-level="12.2" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#download-the-data-3"><i class="fa fa-check"></i><b>12.2</b> Download the data</a></li>
<li class="chapter" data-level="12.3" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#install-transdecoder"><i class="fa fa-check"></i><b>12.3</b> Install Transdecoder</a></li>
<li class="chapter" data-level="12.4" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#run-transdecoder"><i class="fa fa-check"></i><b>12.4</b> Run Transdecoder</a></li>
<li class="chapter" data-level="12.5" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#run-ncbi-blast"><i class="fa fa-check"></i><b>12.5</b> Run NCBI-BLAST</a></li>
<li class="chapter" data-level="12.6" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#what-did-we-find"><i class="fa fa-check"></i><b>12.6</b> What did we find?</a></li>
<li class="chapter" data-level="12.7" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#the-importance-of-genomic-architecture"><i class="fa fa-check"></i><b>12.7</b> The importance of Genomic Architecture</a></li>
<li class="chapter" data-level="12.8" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#exercises-3"><i class="fa fa-check"></i><b>12.8</b> Exercises</a></li>
<li class="chapter" data-level="12.9" data-path="week-10--genomics-is-a-blast.html"><a href="week-10--genomics-is-a-blast.html#creature-of-the-week-9"><i class="fa fa-check"></i><b>12.9</b> Creature of the Week!</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="week-10--genomics-is-a-blast" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">12</span> Week 10- Genomics is a BLAST<a href="week-10--genomics-is-a-blast.html#week-10--genomics-is-a-blast" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Through this course we’ve taken genomic data in the form of whole genome sequences or reduced representation sequences (i.e., RADtags) and searched for correlations between phenotypes (via GWAS) and population structure (Fst Outliers) to identify “candidate SNP” that may have some role in producing the phenotypes we’re interested in.</p>
<p>The next logical step when you’ve identified a candidate SNP is to ask “what’s the function of the gene region near the candidate SNP?” This seems like a straight forward question, as the main goal of many of these studies is to identify the genetic basis of a given phenotype. However, as we dig into this question we will see that it is rather opaque. First we need to consider whether the SNP we identified is “the” SNP itself that is under selection or is the SNP associated through physical linkage (aka hitchhiking), with the gene or gene region that is under selection?</p>
<div id="genome-annotation-and-blast-searches" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> Genome Annotation and BLAST searches<a href="week-10--genomics-is-a-blast.html#genome-annotation-and-blast-searches" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This question can be addressed through whole genome resequencing and identification of regions with reduced heterozygosity which indicate patterns of linked-selection. This is reviewed more fully <a href="http://cooplab.github.io/popgen-notes/#the-effect-of-linked-selection-on-patterns-of-neutral-diversity">here</a></p>
<p>Once a region under selection is identified we can refer to a genome annotation to determine whether this gene region is a coding or non-coding sequence (does it produce a protein or not?) and make some guesses as to the functionalization of this region. Your ability to do this is heavily influenced by the species you’re working on and the degree to which your genome is successfully annotated (if at all). This is described in more detail in the lecture for the week <a href="">here</a></p>
<p>For this weeks exercise we’ll go through a brief illustration of the difficulties in “assigning” putative function to a region surrounding a candidate SNP, using the Fst outliers that we identified in week 7 with the sea cucumber (<strong>Parastichipus californicus</strong>) data published by <a href="http://www2.bio.ulaval.ca/louisbernatchez/pdf/(417)%20Xuereb_Molecular_Ecology_2018.pdf">Xuereb et al.</a></p>
<p>For this data we found 17 Fst outliers (see the pink open circles in the plot below).</p>
<p>I selected a region spanning 20kb around each SNP from the closely related Parastichipus parvimensis genome.This is a relatively large region which should capture any coding sequences that may occur near our SNP. For these regions we will performed a “small scale” annotation by identifying coding regions with the program transdecoder and then running these regions through ncbi-blast which will provide us with a list of similar sequences in other organisms and their gene ids.</p>
<p><img src="figs/week10_figure.png" /></p>
</div>
<div id="download-the-data-3" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Download the data<a href="week-10--genomics-is-a-blast.html#download-the-data-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You’ll download one fasta file that contains the 17 regions in which we found our Fst outlier SNPs.</p>
<div class="sourceCode" id="cb546"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb546-1"><a href="week-10--genomics-is-a-blast.html#cb546-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb546-2"><a href="week-10--genomics-is-a-blast.html#cb546-2" aria-hidden="true" tabindex="-1"></a>wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/week10.tar.gz</span>
<span id="cb546-3"><a href="week-10--genomics-is-a-blast.html#cb546-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb546-4"><a href="week-10--genomics-is-a-blast.html#cb546-4" aria-hidden="true" tabindex="-1"></a>tar -xzvf week10.tar.gz</span></code></pre></div>
</div>
<div id="install-transdecoder" class="section level2 hasAnchor" number="12.3">
<h2><span class="header-section-number">12.3</span> Install Transdecoder<a href="week-10--genomics-is-a-blast.html#install-transdecoder" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ll use transdecoder to identify coding regions in our nucleotide sequences.</p>
<div class="sourceCode" id="cb547"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb547-1"><a href="week-10--genomics-is-a-blast.html#cb547-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb547-2"><a href="week-10--genomics-is-a-blast.html#cb547-2" aria-hidden="true" tabindex="-1"></a>sudo apt update #this just updates our currently installed packages</span>
<span id="cb547-3"><a href="week-10--genomics-is-a-blast.html#cb547-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb547-4"><a href="week-10--genomics-is-a-blast.html#cb547-4" aria-hidden="true" tabindex="-1"></a>sudo apt install transdecoder</span></code></pre></div>
</div>
<div id="run-transdecoder" class="section level2 hasAnchor" number="12.4">
<h2><span class="header-section-number">12.4</span> Run Transdecoder<a href="week-10--genomics-is-a-blast.html#run-transdecoder" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Transdecoder is run in two parts, the first <code>TransDecoder.LongOrfs</code> identifies open read frames or regions that contain either a start or stop codon. This is followed by <code>TransDecoder.Predict</code> which</p>
<div class="sourceCode" id="cb548"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb548-1"><a href="week-10--genomics-is-a-blast.html#cb548-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb548-2"><a href="week-10--genomics-is-a-blast.html#cb548-2" aria-hidden="true" tabindex="-1"></a>TransDecoder.LongOrfs -t candidate_fastas.fa</span>
<span id="cb548-3"><a href="week-10--genomics-is-a-blast.html#cb548-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb548-4"><a href="week-10--genomics-is-a-blast.html#cb548-4" aria-hidden="true" tabindex="-1"></a>TransDecoder.Predict -t candidate_fastas.fa</span></code></pre></div>
<p>This produces an output directory and pulls from that several files with different extensions (.gff3, .bed, .cds, .pep) which correspond to different formats specifically, <code>.gff3</code> is a general feature format file (version 3) and contains information about each of our sequences. The <code>.bed</code> format is a browser extensible data format which describes the chromosome, position and length of our sequences, the <code>.cds</code> contains the nucleotide sequences for the open reading frames, and finally the file we want is the <code>.pep</code> file which contains our open reading frames from our regions converted into protein codes.</p>
</div>
<div id="run-ncbi-blast" class="section level2 hasAnchor" number="12.5">
<h2><span class="header-section-number">12.5</span> Run NCBI-BLAST<a href="week-10--genomics-is-a-blast.html#run-ncbi-blast" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>NCBI-BLAST is commonly called “blast” and stands for Basic Local Alignment Search Tool. This program is often a part of the annotation process as it compares your sequences (or query) to those contained in a database. As such blast results are highly dependent on the database in which you search for results and whether your species of interest (or a close relative) is present in that database. This program can be run in the command line, but in my testing I found it to be incredibly slow on jetstream. So We’re going to run our file on the NCBI-BLAST web browser which should only take a few minutes as we only have 14 protein sequences.</p>
<p>Once you have the <code>.pep</code> file from transdecoder, navigate to the web browser on jetstream (the default is firefox). Search for NCBI-Blast.</p>
<p>We’ll do a protein to protein search (blastp) against the nr database, which contains protein sequences from a variety of taxa, but does not include SRA data (which would include the genome sequence for our sea cucumbers).</p>
</div>
<div id="what-did-we-find" class="section level2 hasAnchor" number="12.6">
<h2><span class="header-section-number">12.6</span> What did we find?<a href="week-10--genomics-is-a-blast.html#what-did-we-find" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Nine out of our 14 protein sequences came back with a blast hit. Many of those that found a hit in Blast were for the other sea cucumber species Parastichipus ja. Many of these sequences are listed as “uncharacterized” or “hypothetical” protein meaning that there has not yet been any downstream classification of these sequences. This is a pretty common result for marine organisms, as they are rarely as highly studied as model organisms such as Drosophila or zebra fish.</p>
</div>
<div id="the-importance-of-genomic-architecture" class="section level2 hasAnchor" number="12.7">
<h2><span class="header-section-number">12.7</span> The importance of Genomic Architecture<a href="week-10--genomics-is-a-blast.html#the-importance-of-genomic-architecture" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Even though we often don’t get satisfying answers about which genes are doing what in Marine Genomics, we can learn a lot about trait evolution from using Fst outliers and tests of Genome Wide Association. These tests give us some idea of the number of genes and their effect on a given trait and that in turn has big implications on how that trait can respond to evolution. And that in turn has implications for climate change, conservation, and biodiversity.</p>
</div>
<div id="exercises-3" class="section level2 hasAnchor" number="12.8">
<h2><span class="header-section-number">12.8</span> Exercises<a href="week-10--genomics-is-a-blast.html#exercises-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<h1 id="practice-questions-4" class="hasAnchor">Practice Questions:<a href="week-10--genomics-is-a-blast.html#practice-questions-4" class="anchor-section" aria-label="Anchor link to header"></a></h1>
</blockquote>
<blockquote>
<ol style="list-style-type: decimal">
<li>Instead of doing a blast search for the protein translated sequences, do one for the nucleotide sequences. Use the original `candidate_fastas.fa file, and do a blastn for somewhat similar sequences. How do these results differ from our search with the translated open reading frames? Why? Try again as a megablast for highly similar sequences (you can do this by editing your original search). How do these differ?</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: SeaGreen;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="week-10--genomics-is-a-blast.html#cb549-1" aria-hidden="true" tabindex="-1"></a><span class="co">#coming soon :) </span></span></code></pre></div>
</p>
</details>
<blockquote>
<ol start="2" style="list-style-type: decimal">
<li>Now use the <code>.cds</code> file from transdecoder which contains the open reading frame as nucleotides.</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: SeaGreen;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb550"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb550-1"><a href="week-10--genomics-is-a-blast.html#cb550-1" aria-hidden="true" tabindex="-1"></a><span class="co">#coming soon :) </span></span></code></pre></div>
</p>
</details>
</div>
<div id="creature-of-the-week-9" class="section level2 hasAnchor" number="12.9">
<h2><span class="header-section-number">12.9</span> Creature of the Week!<a href="week-10--genomics-is-a-blast.html#creature-of-the-week-9" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure">
<img src="figs/creatures/vaquitaporpoise.jpg" alt="" />
<p class="caption">Vaquita Porpoise (image credit: Brendan Hunter)</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="week-9--population-structure-using-ngsadmix.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Test.pdf", "Test.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
